// @salt-ds/core/dist-cjs/dialog/DialogContent.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
var makePrefixer = require('../utils/makePrefixer.js');
var useIsomorphicLayoutEffect = require('../utils/useIsomorphicLayoutEffect.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
var useResizeObserver = require('../utils/useResizeObserver.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var DialogContent$1 = require('./DialogContent.css.js');
var DialogContext = require('./DialogContext.js');

const withBaseName = makePrefixer.makePrefixer("saltDialogContent");
const DialogContent = React.forwardRef(
  function DialogContent2(props, ref) {
    const { children, className, ...rest } = props;
    const [canScrollUp, setCanScrollUp] = React.useState(false);
    const [canScrollDown, setCanScrollDown] = React.useState(true);
    const [isOverflowing, setIsOverflowing] = React.useState(false);
    const divRef = React.useRef(null);
    const handleScroll = () => {
      targetWindow == null ? void 0 : targetWindow.requestAnimationFrame(() => {
        const container = divRef.current;
        if (!container) return;
        setCanScrollUp(container.scrollTop > 0);
        setCanScrollDown(
          container.scrollHeight - container.scrollTop - container.clientHeight > 1
        );
      });
    };
    const targetWindow = window.useWindow();
    styles.useComponentCssInjection({
      testId: "salt-dialog-content",
      css: DialogContent$1,
      window: targetWindow
    });
    const checkOverflow = React.useCallback(() => {
      if (!divRef.current) return;
      setIsOverflowing(
        divRef.current.scrollWidth > divRef.current.offsetWidth || divRef.current.scrollHeight > divRef.current.offsetHeight
      );
    }, []);
    useResizeObserver.useResizeObserver({ ref: divRef, onResize: checkOverflow });
    useIsomorphicLayoutEffect.useIsomorphicLayoutEffect(() => {
      checkOverflow();
    }, [checkOverflow]);
    const { dialogId, id: headerId } = DialogContext.useDialogContext();
    const overflowProps = isOverflowing ? {
      role: "region",
      tabIndex: 0,
      "aria-labelledby": headerId ?? dialogId
    } : {};
    return /* @__PURE__ */ jsxRuntime.jsx("div", { className: clsx.clsx(withBaseName(), className), ...rest, ref, children: /* @__PURE__ */ jsxRuntime.jsx(
      "div",
      {
        onScrollCapture: handleScroll,
        ref: divRef,
        className: clsx.clsx(withBaseName("inner"), {
          [withBaseName("overflow")]: isOverflowing,
          [withBaseName("scrollTop")]: isOverflowing && canScrollUp,
          [withBaseName("scrollBottom")]: isOverflowing && canScrollDown
        }),
        ...overflowProps,
        children
      }
    ) });
  }
);

exports.DialogContent = DialogContent;
