// @salt-ds/core/dist-es/vertical-navigation/VerticalNavigationSubMenu.js
import { jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef, useState } from 'react';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { SubMenuProvider } from './SubMenuContext.js';
import css_248z from './VerticalNavigationSubMenu.css.js';

const withBaseName = makePrefixer("saltVerticalNavigationSubMenu");
const VerticalNavigationSubMenu = forwardRef(function VerticalNavigationSubMenu2(props, ref) {
  const { children, className, ...rest } = props;
  const targetWindow = useWindow();
  useComponentCssInjection({
    testId: "salt-vertical-navigation-sub-menu",
    css: css_248z,
    window: targetWindow
  });
  const [directIcons, setDirectIcons] = useState([]);
  return /* @__PURE__ */ jsx(SubMenuProvider, { directIcons, setDirectIcons, children: /* @__PURE__ */ jsx(
    "ul",
    {
      ref,
      className: clsx(withBaseName(), className),
      "data-has-direct-icons": directIcons.length > 0,
      ...rest,
      children
    }
  ) });
});

export { VerticalNavigationSubMenu };
