// @salt-ds/core/dist-es/vertical-navigation/VerticalNavigationItemContent.js
import { jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef, useRef, useState, useEffect } from 'react';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import { useForkRef } from '../utils/useForkRef.js';
import { useId } from '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { useSubMenuContext } from './SubMenuContext.js';
import { useVerticalNavigationItem } from './VerticalNavigationItem.js';
import css_248z from './VerticalNavigationItemContent.css.js';

const withBaseName = makePrefixer("saltVerticalNavigationItemContent");
const VerticalNavigationItemContent = forwardRef(function VerticalNavigationItemContent2(props, ref) {
  const { children, className, style, ...rest } = props;
  const targetWindow = useWindow();
  useComponentCssInjection({
    testId: "salt-vertical-navigation-item-content",
    css: css_248z,
    window: targetWindow
  });
  const containerRef = useRef(null);
  const handleRef = useForkRef(ref, containerRef);
  const { active, focusVisible } = useVerticalNavigationItem();
  const { setDirectIcons, iconPaddingCount } = useSubMenuContext();
  const [hasIcon, setHasIcon] = useState(false);
  const iconId = useId();
  useEffect(() => {
    const checkForIcons = () => {
      var _a;
      const iconElement = (_a = containerRef.current) == null ? void 0 : _a.querySelector(
        ".saltIcon ~ .saltVerticalNavigationItemLabel"
      );
      setHasIcon(Boolean(iconElement));
      if (iconElement && iconId && setDirectIcons) {
        setDirectIcons((icons) => {
          if (!icons.includes(iconId)) {
            return icons.concat(iconId);
          }
          return icons;
        });
      } else if (!iconElement && iconId && setDirectIcons) {
        setDirectIcons((icons) => icons.filter((id) => id !== iconId));
      }
    };
    const mutationObserver = new MutationObserver(() => {
      checkForIcons();
    });
    checkForIcons();
    if (containerRef.current) {
      mutationObserver.observe(containerRef.current, {
        childList: true,
        subtree: true
      });
    }
    return () => {
      mutationObserver.disconnect();
    };
  }, [iconId, setDirectIcons]);
  return /* @__PURE__ */ jsx(
    "span",
    {
      ref: handleRef,
      className: clsx(
        withBaseName(),
        {
          [withBaseName("active")]: active,
          [withBaseName("focused")]: focusVisible
        },
        className
      ),
      style: {
        "--saltVerticalNavigationItem-iconPaddingMultiplier": hasIcon ? iconPaddingCount - 1 : iconPaddingCount,
        ...style
      },
      ...rest,
      children
    }
  );
});

export { VerticalNavigationItemContent };
