// @salt-ds/core/dist-es/vertical-navigation/VerticalNavigation.js
import { jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef, useState } from 'react';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { SubMenuProvider } from './SubMenuContext.js';
import css_248z from './VerticalNavigation.css.js';

const withBaseName = makePrefixer("saltVerticalNavigation");
const VerticalNavigation = forwardRef(function VerticalNavigation2(props, ref) {
  const { appearance = "bordered", children, className, ...rest } = props;
  const targetWindow = useWindow();
  useComponentCssInjection({
    testId: "salt-vertical-navigation",
    css: css_248z,
    window: targetWindow
  });
  const [directIcons, setDirectIcons] = useState([]);
  return /* @__PURE__ */ jsx(SubMenuProvider, { directIcons, setDirectIcons, children: /* @__PURE__ */ jsx(
    "nav",
    {
      ref,
      className: clsx(withBaseName(), withBaseName(appearance), className),
      ...rest,
      children: /* @__PURE__ */ jsx("ol", { "data-has-direct-icons": directIcons.length > 0, children })
    }
  ) });
});

export { VerticalNavigation };
