// @salt-ds/core/dist-es/vertical-navigation/VerticalNavigationItemTrigger.js
import { jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef, useRef } from 'react';
import { makePrefixer } from '../utils/makePrefixer.js';
import { renderProps } from '../utils/renderProps.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { useVerticalNavigationItem } from './VerticalNavigationItem.js';
import css_248z from './VerticalNavigationItemTrigger.css.js';

function ItemAction(props) {
  return renderProps("a", props);
}
const withBaseName = makePrefixer("saltVerticalNavigationItemTrigger");
const VerticalNavigationItemTrigger = forwardRef(function VerticalNavigationItemTrigger2(props, ref) {
  const {
    className,
    children,
    render,
    href,
    onFocus,
    onBlur,
    onMouseDown,
    ...rest
  } = props;
  const targetWindow = useWindow();
  useComponentCssInjection({
    testId: "salt-vertical-navigation-item-trigger",
    css: css_248z,
    window: targetWindow
  });
  const isLink = href != null;
  const { active, setFocusVisible } = useVerticalNavigationItem();
  const wasMouseDownRef = useRef(false);
  const handleFocus = (event) => {
    if (!wasMouseDownRef.current) {
      setFocusVisible(true);
    }
    wasMouseDownRef.current = false;
    onFocus == null ? void 0 : onFocus(event);
  };
  const handleBlur = (event) => {
    setFocusVisible(false);
    onBlur == null ? void 0 : onBlur(event);
  };
  const handleMouseDown = (event) => {
    wasMouseDownRef.current = true;
    onMouseDown == null ? void 0 : onMouseDown(event);
  };
  return /* @__PURE__ */ jsx(
    ItemAction,
    {
      className: clsx(withBaseName(), className),
      href,
      "aria-current": active ? "page" : void 0,
      render: render ?? (isLink ? void 0 : /* @__PURE__ */ jsx("button", { type: "button" })),
      ref,
      onFocus: handleFocus,
      onBlur: handleBlur,
      onMouseDown: handleMouseDown,
      ...rest,
      children
    }
  );
});

export { VerticalNavigationItemTrigger };
