// @salt-ds/core/dist-es/vertical-navigation/VerticalNavigationItem.js
import { jsx } from 'react/jsx-runtime';
import { clsx } from 'clsx';
import { forwardRef, useState, useMemo, useContext } from 'react';
import { createContext } from '../utils/createContext.js';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { useSubMenuContext } from './SubMenuContext.js';

const withBaseName = makePrefixer("saltVerticalNavigationItem");
const VerticalNavigationItemContext = createContext(
  "saltVerticalNavigationItemContext",
  {
    active: false,
    focusVisible: false,
    setFocusVisible: () => {
    }
  }
);
function useVerticalNavigationItem() {
  return useContext(VerticalNavigationItemContext);
}
const VerticalNavigationItem = forwardRef(function VerticalNavigationItem2(props, ref) {
  const { children, className, active = false, style, ...rest } = props;
  const { depth } = useSubMenuContext();
  const [focusVisible, setFocusVisible] = useState(false);
  const context = useMemo(
    () => ({
      active,
      focusVisible,
      setFocusVisible
    }),
    [active, focusVisible]
  );
  return /* @__PURE__ */ jsx(VerticalNavigationItemContext.Provider, { value: context, children: /* @__PURE__ */ jsx(
    "li",
    {
      ref,
      className: clsx(withBaseName(), className),
      style: { "--verticalNavigationItem-depth": depth, ...style },
      ...rest,
      children
    }
  ) });
});

export { VerticalNavigationItem, useVerticalNavigationItem };
