// @salt-ds/core/dist-cjs/vertical-navigation/VerticalNavigationItemTrigger.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
var makePrefixer = require('../utils/makePrefixer.js');
var renderProps = require('../utils/renderProps.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var VerticalNavigationItem = require('./VerticalNavigationItem.js');
var VerticalNavigationItemTrigger$1 = require('./VerticalNavigationItemTrigger.css.js');

function ItemAction(props) {
  return renderProps.renderProps("a", props);
}
const withBaseName = makePrefixer.makePrefixer("saltVerticalNavigationItemTrigger");
const VerticalNavigationItemTrigger = React.forwardRef(function VerticalNavigationItemTrigger2(props, ref) {
  const {
    className,
    children,
    render,
    href,
    onFocus,
    onBlur,
    onMouseDown,
    ...rest
  } = props;
  const targetWindow = window.useWindow();
  styles.useComponentCssInjection({
    testId: "salt-vertical-navigation-item-trigger",
    css: VerticalNavigationItemTrigger$1,
    window: targetWindow
  });
  const isLink = href != null;
  const { active, setFocusVisible } = VerticalNavigationItem.useVerticalNavigationItem();
  const wasMouseDownRef = React.useRef(false);
  const handleFocus = (event) => {
    if (!wasMouseDownRef.current) {
      setFocusVisible(true);
    }
    wasMouseDownRef.current = false;
    onFocus == null ? void 0 : onFocus(event);
  };
  const handleBlur = (event) => {
    setFocusVisible(false);
    onBlur == null ? void 0 : onBlur(event);
  };
  const handleMouseDown = (event) => {
    wasMouseDownRef.current = true;
    onMouseDown == null ? void 0 : onMouseDown(event);
  };
  return /* @__PURE__ */ jsxRuntime.jsx(
    ItemAction,
    {
      className: clsx.clsx(withBaseName(), className),
      href,
      "aria-current": active ? "page" : void 0,
      render: render ?? (isLink ? void 0 : /* @__PURE__ */ jsxRuntime.jsx("button", { type: "button" })),
      ref,
      onFocus: handleFocus,
      onBlur: handleBlur,
      onMouseDown: handleMouseDown,
      ...rest,
      children
    }
  );
});

exports.VerticalNavigationItemTrigger = VerticalNavigationItemTrigger;
