// @salt-ds/core/dist-cjs/vertical-navigation/VerticalNavigationSubMenu.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var SubMenuContext = require('./SubMenuContext.js');
var VerticalNavigationSubMenu$1 = require('./VerticalNavigationSubMenu.css.js');

const withBaseName = makePrefixer.makePrefixer("saltVerticalNavigationSubMenu");
const VerticalNavigationSubMenu = React.forwardRef(function VerticalNavigationSubMenu2(props, ref) {
  const { children, className, ...rest } = props;
  const targetWindow = window.useWindow();
  styles.useComponentCssInjection({
    testId: "salt-vertical-navigation-sub-menu",
    css: VerticalNavigationSubMenu$1,
    window: targetWindow
  });
  const [directIcons, setDirectIcons] = React.useState([]);
  return /* @__PURE__ */ jsxRuntime.jsx(SubMenuContext.SubMenuProvider, { directIcons, setDirectIcons, children: /* @__PURE__ */ jsxRuntime.jsx(
    "ul",
    {
      ref,
      className: clsx.clsx(withBaseName(), className),
      "data-has-direct-icons": directIcons.length > 0,
      ...rest,
      children
    }
  ) });
});

exports.VerticalNavigationSubMenu = VerticalNavigationSubMenu;
