// @salt-ds/core/dist-cjs/vertical-navigation/VerticalNavigationItemContent.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
var useForkRef = require('../utils/useForkRef.js');
var useId = require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var SubMenuContext = require('./SubMenuContext.js');
var VerticalNavigationItem = require('./VerticalNavigationItem.js');
var VerticalNavigationItemContent$1 = require('./VerticalNavigationItemContent.css.js');

const withBaseName = makePrefixer.makePrefixer("saltVerticalNavigationItemContent");
const VerticalNavigationItemContent = React.forwardRef(function VerticalNavigationItemContent2(props, ref) {
  const { children, className, style, ...rest } = props;
  const targetWindow = window.useWindow();
  styles.useComponentCssInjection({
    testId: "salt-vertical-navigation-item-content",
    css: VerticalNavigationItemContent$1,
    window: targetWindow
  });
  const containerRef = React.useRef(null);
  const handleRef = useForkRef.useForkRef(ref, containerRef);
  const { active, focusVisible } = VerticalNavigationItem.useVerticalNavigationItem();
  const { setDirectIcons, iconPaddingCount } = SubMenuContext.useSubMenuContext();
  const [hasIcon, setHasIcon] = React.useState(false);
  const iconId = useId.useId();
  React.useEffect(() => {
    const checkForIcons = () => {
      var _a;
      const iconElement = (_a = containerRef.current) == null ? void 0 : _a.querySelector(
        ".saltIcon ~ .saltVerticalNavigationItemLabel"
      );
      setHasIcon(Boolean(iconElement));
      if (iconElement && iconId && setDirectIcons) {
        setDirectIcons((icons) => {
          if (!icons.includes(iconId)) {
            return icons.concat(iconId);
          }
          return icons;
        });
      } else if (!iconElement && iconId && setDirectIcons) {
        setDirectIcons((icons) => icons.filter((id) => id !== iconId));
      }
    };
    const mutationObserver = new MutationObserver(() => {
      checkForIcons();
    });
    checkForIcons();
    if (containerRef.current) {
      mutationObserver.observe(containerRef.current, {
        childList: true,
        subtree: true
      });
    }
    return () => {
      mutationObserver.disconnect();
    };
  }, [iconId, setDirectIcons]);
  return /* @__PURE__ */ jsxRuntime.jsx(
    "span",
    {
      ref: handleRef,
      className: clsx.clsx(
        withBaseName(),
        {
          [withBaseName("active")]: active,
          [withBaseName("focused")]: focusVisible
        },
        className
      ),
      style: {
        "--saltVerticalNavigationItem-iconPaddingMultiplier": hasIcon ? iconPaddingCount - 1 : iconPaddingCount,
        ...style
      },
      ...rest,
      children
    }
  );
});

exports.VerticalNavigationItemContent = VerticalNavigationItemContent;
