// @salt-ds/core/dist-cjs/vertical-navigation/VerticalNavigation.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var SubMenuContext = require('./SubMenuContext.js');
var VerticalNavigation$1 = require('./VerticalNavigation.css.js');

const withBaseName = makePrefixer.makePrefixer("saltVerticalNavigation");
const VerticalNavigation = React.forwardRef(function VerticalNavigation2(props, ref) {
  const { appearance = "bordered", children, className, ...rest } = props;
  const targetWindow = window.useWindow();
  styles.useComponentCssInjection({
    testId: "salt-vertical-navigation",
    css: VerticalNavigation$1,
    window: targetWindow
  });
  const [directIcons, setDirectIcons] = React.useState([]);
  return /* @__PURE__ */ jsxRuntime.jsx(SubMenuContext.SubMenuProvider, { directIcons, setDirectIcons, children: /* @__PURE__ */ jsxRuntime.jsx(
    "nav",
    {
      ref,
      className: clsx.clsx(withBaseName(), withBaseName(appearance), className),
      ...rest,
      children: /* @__PURE__ */ jsxRuntime.jsx("ol", { "data-has-direct-icons": directIcons.length > 0, children })
    }
  ) });
});

exports.VerticalNavigation = VerticalNavigation;
