// @salt-ds/core/dist-cjs/vertical-navigation/VerticalNavigationItem.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var clsx = require('clsx');
var React = require('react');
var createContext = require('../utils/createContext.js');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var SubMenuContext = require('./SubMenuContext.js');

const withBaseName = makePrefixer.makePrefixer("saltVerticalNavigationItem");
const VerticalNavigationItemContext = createContext.createContext(
  "saltVerticalNavigationItemContext",
  {
    active: false,
    focusVisible: false,
    setFocusVisible: () => {
    }
  }
);
function useVerticalNavigationItem() {
  return React.useContext(VerticalNavigationItemContext);
}
const VerticalNavigationItem = React.forwardRef(function VerticalNavigationItem2(props, ref) {
  const { children, className, active = false, style, ...rest } = props;
  const { depth } = SubMenuContext.useSubMenuContext();
  const [focusVisible, setFocusVisible] = React.useState(false);
  const context = React.useMemo(
    () => ({
      active,
      focusVisible,
      setFocusVisible
    }),
    [active, focusVisible]
  );
  return /* @__PURE__ */ jsxRuntime.jsx(VerticalNavigationItemContext.Provider, { value: context, children: /* @__PURE__ */ jsxRuntime.jsx(
    "li",
    {
      ref,
      className: clsx.clsx(withBaseName(), className),
      style: { "--verticalNavigationItem-depth": depth, ...style },
      ...rest,
      children
    }
  ) });
});

exports.VerticalNavigationItem = VerticalNavigationItem;
exports.useVerticalNavigationItem = useVerticalNavigationItem;
