// @salt-ds/core/dist-es/switch/Switch.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { CheckmarkSolidIcon, CheckmarkIcon } from '@salt-ds/icons';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef } from 'react';
import '../form-field-context/FormFieldContext.js';
import { useFormFieldProps } from '../form-field-context/useFormFieldProps.js';
import { makePrefixer } from '../utils/makePrefixer.js';
import { useControlled } from '../utils/useControlled.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import css_248z from './Switch.css.js';

const withBaseName = makePrefixer("saltSwitch");
const Switch = forwardRef(
  function Switch2(props, ref) {
    const {
      checked: checkedProp,
      className,
      defaultChecked,
      disabled: disabledProp,
      inputProps = {},
      label,
      name,
      onBlur,
      onChange,
      onFocus,
      value,
      readOnly: readOnlyProp,
      ...rest
    } = props;
    const targetWindow = useWindow();
    useComponentCssInjection({
      testId: "salt-switch",
      css: css_248z,
      window: targetWindow
    });
    const {
      "aria-describedby": inputDescribedBy,
      "aria-labelledby": inputLabelledBy,
      className: inputClassName,
      onChange: inputOnChange,
      ...restInputProps
    } = inputProps;
    const [checked, setChecked] = useControlled({
      controlled: checkedProp,
      default: Boolean(defaultChecked),
      name: "Switch",
      state: "checked"
    });
    const {
      a11yProps: formFieldA11yProps,
      disabled: formFieldDisabled,
      readOnly: formFieldReadOnly
    } = useFormFieldProps();
    const disabled = formFieldDisabled || disabledProp;
    const readOnly = formFieldReadOnly || readOnlyProp;
    const handleChange = (event) => {
      if (event.nativeEvent.defaultPrevented || readOnly) {
        return;
      }
      const value2 = event.target.checked;
      setChecked(value2);
      onChange == null ? void 0 : onChange(event);
      inputOnChange == null ? void 0 : inputOnChange(event);
    };
    return /* @__PURE__ */ jsxs(
      "label",
      {
        className: clsx(
          withBaseName(),
          {
            [withBaseName("disabled")]: disabled,
            [withBaseName("checked")]: checked,
            [withBaseName("readOnly")]: readOnly
          },
          className
        ),
        ref,
        ...rest,
        children: [
          /* @__PURE__ */ jsx(
            "input",
            {
              "aria-readonly": readOnly || void 0,
              "aria-describedby": clsx(formFieldA11yProps == null ? void 0 : formFieldA11yProps["aria-describedby"], inputDescribedBy) || void 0,
              "aria-labelledby": clsx(formFieldA11yProps == null ? void 0 : formFieldA11yProps["aria-labelledby"], inputLabelledBy) || void 0,
              name,
              value,
              checked,
              className: clsx(withBaseName("input"), inputClassName),
              defaultChecked,
              disabled,
              onBlur,
              onChange: handleChange,
              onFocus,
              type: "checkbox",
              role: "switch",
              ...restInputProps
            }
          ),
          /* @__PURE__ */ jsx("span", { className: withBaseName("track"), children: /* @__PURE__ */ jsxs("span", { className: withBaseName("thumb"), children: [
            checked && !readOnly && /* @__PURE__ */ jsx(
              CheckmarkSolidIcon,
              {
                "aria-hidden": true,
                className: withBaseName("icon")
              }
            ),
            checked && readOnly && /* @__PURE__ */ jsx(CheckmarkIcon, { "aria-hidden": true, className: withBaseName("icon") })
          ] }) }),
          label && /* @__PURE__ */ jsx("span", { className: withBaseName("label"), children: label })
        ]
      }
    );
  }
);

export { Switch };
