// @salt-ds/core/dist-es/status-indicator/StatusIndicator.js
import { jsx } from 'react/jsx-runtime';
import { DEFAULT_ICON_SIZE } from '@salt-ds/icons';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef } from 'react';
import { useIcon } from '../semantic-icon-provider/SemanticIconProvider.js';
import { capitalize } from '../utils/capitalize.js';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import css_248z from './StatusIndicator.css.js';

const withBaseName = makePrefixer("saltStatusIndicator");
const StatusIndicator = forwardRef(
  function StatusIndicator2({ className, status, size = DEFAULT_ICON_SIZE, ...restProps }, ref) {
    const targetWindow = useWindow();
    useComponentCssInjection({
      testId: "salt-status-indicator",
      css: css_248z,
      window: targetWindow
    });
    const icons = useIcon();
    const titleCaseStatus = capitalize(status);
    const iconKey = `${titleCaseStatus}Icon`;
    const IconComponent = icons[iconKey];
    const ariaLabel = status;
    if (IconComponent === void 0) {
      return null;
    }
    return /* @__PURE__ */ jsx(
      IconComponent,
      {
        className: clsx(withBaseName(), withBaseName(status), className),
        size,
        "aria-label": ariaLabel,
        ...restProps,
        ref
      }
    );
  }
);

export { StatusIndicator };
