// @salt-ds/core/dist-es/dialog/DialogContent.js
import { jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef, useState, useRef, useCallback } from 'react';
import { makePrefixer } from '../utils/makePrefixer.js';
import { useIsomorphicLayoutEffect } from '../utils/useIsomorphicLayoutEffect.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import { useResizeObserver } from '../utils/useResizeObserver.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import css_248z from './DialogContent.css.js';
import { useDialogContext } from './DialogContext.js';

const withBaseName = makePrefixer("saltDialogContent");
const DialogContent = forwardRef(
  function DialogContent2(props, ref) {
    const { children, className, ...rest } = props;
    const [canScrollUp, setCanScrollUp] = useState(false);
    const [canScrollDown, setCanScrollDown] = useState(true);
    const [isOverflowing, setIsOverflowing] = useState(false);
    const divRef = useRef(null);
    const handleScroll = () => {
      targetWindow == null ? void 0 : targetWindow.requestAnimationFrame(() => {
        const container = divRef.current;
        if (!container) return;
        setCanScrollUp(container.scrollTop > 0);
        setCanScrollDown(
          container.scrollHeight - container.scrollTop - container.clientHeight > 1
        );
      });
    };
    const targetWindow = useWindow();
    useComponentCssInjection({
      testId: "salt-dialog-content",
      css: css_248z,
      window: targetWindow
    });
    const checkOverflow = useCallback(() => {
      if (!divRef.current) return;
      setIsOverflowing(
        divRef.current.scrollWidth > divRef.current.offsetWidth || divRef.current.scrollHeight > divRef.current.offsetHeight
      );
    }, []);
    useResizeObserver({ ref: divRef, onResize: checkOverflow });
    useIsomorphicLayoutEffect(() => {
      checkOverflow();
    }, [checkOverflow]);
    const { dialogId, id: headerId } = useDialogContext();
    const overflowProps = isOverflowing ? {
      role: "region",
      tabIndex: 0,
      "aria-labelledby": headerId ?? dialogId
    } : {};
    return /* @__PURE__ */ jsx("div", { className: clsx(withBaseName(), className), ...rest, ref, children: /* @__PURE__ */ jsx(
      "div",
      {
        onScrollCapture: handleScroll,
        ref: divRef,
        className: clsx(withBaseName("inner"), {
          [withBaseName("overflow")]: isOverflowing,
          [withBaseName("scrollTop")]: isOverflowing && canScrollUp,
          [withBaseName("scrollBottom")]: isOverflowing && canScrollDown
        }),
        ...overflowProps,
        children
      }
    ) });
  }
);

export { DialogContent };
