// @salt-ds/core/dist-cjs/collapsible/Collapsible.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var clsx = require('clsx');
var React = require('react');
var makePrefixer = require('../utils/makePrefixer.js');
var useControlled = require('../utils/useControlled.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var CollapsibleContext = require('./CollapsibleContext.js');

const withBaseName = makePrefixer.makePrefixer("saltCollapsible");
const Collapsible = React.forwardRef(
  function Collapsible2(props, ref) {
    const {
      className,
      open: openProp,
      defaultOpen,
      onOpenChange,
      ...rest
    } = props;
    const [open, setOpenState] = useControlled.useControlled({
      default: Boolean(defaultOpen),
      controlled: openProp,
      name: "Collapsible",
      state: "open"
    });
    const [panelId, setPanelId] = React.useState(void 0);
    const setOpen = React.useCallback(
      (event, newOpen) => {
        setOpenState(newOpen);
        onOpenChange == null ? void 0 : onOpenChange(event, newOpen);
      },
      [onOpenChange]
    );
    const contextValue = React.useMemo(
      () => ({
        open,
        setOpen,
        panelId,
        setPanelId
      }),
      [open, setOpen, panelId]
    );
    return /* @__PURE__ */ jsxRuntime.jsx(CollapsibleContext.CollapsibleContext.Provider, { value: contextValue, children: /* @__PURE__ */ jsxRuntime.jsx("div", { className: clsx.clsx(withBaseName(), className), ref, ...rest }) });
  }
);

exports.Collapsible = Collapsible;
