// @salt-ds/core/dist-cjs/collapsible/CollapsiblePanel.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
var useId = require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var CollapsibleContext = require('./CollapsibleContext.js');
var CollapsiblePanel$1 = require('./CollapsiblePanel.css.js');

const withBaseName = makePrefixer.makePrefixer("saltCollapsiblePanel");
const CollapsiblePanel = React.forwardRef((props, ref) => {
  const { children, className, id: idProp, ...rest } = props;
  const targetWindow = window.useWindow();
  styles.useComponentCssInjection({
    testId: "salt-collapsible-panel",
    css: CollapsiblePanel$1,
    window: targetWindow
  });
  const id = useId.useId(idProp);
  const { open, setPanelId } = CollapsibleContext.useCollapsibleContext();
  React.useEffect(() => {
    if (id) {
      setPanelId == null ? void 0 : setPanelId(id);
    }
  }, [id, setPanelId]);
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: clsx.clsx(withBaseName(), className),
      id,
      "aria-hidden": !open ? "true" : void 0,
      hidden: !open,
      ref,
      ...rest,
      children: /* @__PURE__ */ jsxRuntime.jsx("div", { className: withBaseName("inner"), children })
    }
  );
});

exports.CollapsiblePanel = CollapsiblePanel;
