// @salt-ds/core/dist-cjs/kbd/Kbd.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var Kbd$1 = require('./Kbd.css.js');

const withBaseName = makePrefixer.makePrefixer("saltKbd");
const Kbd = React.forwardRef(
  function Kbd2(props, ref) {
    const { children, className, variant = "primary", ...rest } = props;
    const targetWindow = window.useWindow();
    styles.useComponentCssInjection({
      testId: "salt-kbd",
      css: Kbd$1,
      window: targetWindow
    });
    return /* @__PURE__ */ jsxRuntime.jsx(
      "kbd",
      {
        ref,
        className: clsx.clsx(withBaseName(), withBaseName(variant), className),
        ...rest,
        children
      }
    );
  }
);

exports.Kbd = Kbd;
