// @salt-ds/core/dist-cjs/salt-provider/SaltProvider.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
require('../aria-announcer/AriaAnnouncerContext.js');
var AriaAnnouncerProvider = require('../aria-announcer/AriaAnnouncerProvider.js');
var BreakpointProvider = require('../breakpoints/BreakpointProvider.js');
var Breakpoints = require('../breakpoints/Breakpoints.js');
var ViewportProvider = require('../viewport/ViewportProvider.js');
var ProviderContext = require('./ProviderContext.js');
var SaltProvider$1 = require('./SaltProvider.css.js');
var ThemeApplicator = require('./ThemeApplicator.js');

const DEFAULT_DENSITY = "medium";
const DEFAULT_THEME_NAME = "salt-theme";
const DEFAULT_THEME_NAME_NEXT = "salt-theme-next";
const DEFAULT_MODE = "light";
const DEFAULT_CORNER = "sharp";
const DEFAULT_HEADING_FONT = "Open Sans";
const DEFAULT_ACCENT = "blue";
const DEFAULT_ACTION_FONT = "Open Sans";
const DensityContext = React.createContext(DEFAULT_DENSITY);
const ThemeContext = React.createContext({
  theme: "",
  mode: DEFAULT_MODE,
  themeNext: false,
  corner: DEFAULT_CORNER,
  UNSTABLE_corner: DEFAULT_CORNER,
  headingFont: DEFAULT_HEADING_FONT,
  UNSTABLE_headingFont: DEFAULT_HEADING_FONT,
  accent: DEFAULT_ACCENT,
  UNSTABLE_accent: DEFAULT_ACCENT,
  actionFont: DEFAULT_ACTION_FONT,
  UNSTABLE_actionFont: DEFAULT_ACTION_FONT
});
const BreakpointContext = React.createContext(Breakpoints.DEFAULT_BREAKPOINTS);
function InternalSaltProvider({
  applyClassesTo: applyClassesToProp,
  children,
  density: densityProp,
  theme: themeProp,
  mode: modeProp,
  breakpoints: breakpointsProp,
  themeNext,
  corner: cornerProp,
  headingFont: headingFontProp,
  accent: accentProp,
  actionFont: actionFontProp
}) {
  const prevProvider = React.useContext(ProviderContext.ProviderContext);
  const inheritedDensity = React.useContext(DensityContext);
  const {
    theme: inheritedTheme,
    mode: inheritedMode,
    corner: inheritedCorner,
    headingFont: inheritedHeadingFont,
    accent: inheritedAccent,
    actionFont: inheritedActionFont
  } = React.useContext(ThemeContext);
  const isRootProvider = prevProvider === null;
  const density = densityProp ?? inheritedDensity ?? DEFAULT_DENSITY;
  const inheritedThemeName = inheritedTheme === "" ? DEFAULT_THEME_NAME : inheritedTheme;
  const themeName = themeProp ?? inheritedThemeName;
  const mode = modeProp ?? inheritedMode;
  const breakpoints = breakpointsProp ?? Breakpoints.DEFAULT_BREAKPOINTS;
  const corner = cornerProp ?? inheritedCorner ?? DEFAULT_CORNER;
  const headingFont = headingFontProp ?? inheritedHeadingFont ?? DEFAULT_HEADING_FONT;
  const accent = accentProp ?? inheritedAccent ?? DEFAULT_ACCENT;
  const actionFont = actionFontProp ?? inheritedActionFont ?? DEFAULT_ACTION_FONT;
  const targetWindow = window.useWindow();
  styles.useComponentCssInjection({
    testId: "salt-provider",
    css: SaltProvider$1,
    window: targetWindow
  });
  const themeContextValue = React.useMemo(
    () => ({
      theme: themeName,
      mode,
      window: targetWindow,
      themeNext: Boolean(themeNext),
      corner,
      headingFont,
      accent,
      actionFont,
      // Backward compatibility
      UNSTABLE_corner: corner,
      UNSTABLE_headingFont: headingFont,
      UNSTABLE_accent: accent,
      UNSTABLE_actionFont: actionFont
    }),
    [
      themeName,
      mode,
      targetWindow,
      themeNext,
      corner,
      headingFont,
      accent,
      actionFont
    ]
  );
  const matchedBreakpoints = BreakpointProvider.useMatchedBreakpoints(breakpoints);
  const saltProvider = /* @__PURE__ */ jsxRuntime.jsx(ProviderContext.ProviderContext.Provider, { value: { targetWindow }, children: /* @__PURE__ */ jsxRuntime.jsx(DensityContext.Provider, { value: density, children: /* @__PURE__ */ jsxRuntime.jsx(ThemeContext.Provider, { value: themeContextValue, children: /* @__PURE__ */ jsxRuntime.jsx(BreakpointProvider.BreakpointProvider, { matchedBreakpoints, children: /* @__PURE__ */ jsxRuntime.jsx(BreakpointContext.Provider, { value: breakpoints, children: /* @__PURE__ */ jsxRuntime.jsx(ViewportProvider.ViewportProvider, { children: /* @__PURE__ */ jsxRuntime.jsx(
    ThemeApplicator.ThemeApplicator,
    {
      applyClassesTo: applyClassesToProp ?? (isRootProvider ? "root" : "scope"),
      density,
      theme: clsx.clsx(
        DEFAULT_THEME_NAME,
        { [DEFAULT_THEME_NAME_NEXT]: themeNext },
        themeProp
      ),
      mode,
      accent: themeNext ? accent : void 0,
      actionFont: themeNext ? actionFont : void 0,
      headingFont: themeNext ? headingFont : void 0,
      corner: themeNext ? corner : void 0,
      children
    }
  ) }) }) }) }) }) });
  if (isRootProvider) {
    return /* @__PURE__ */ jsxRuntime.jsx(AriaAnnouncerProvider.AriaAnnouncerProvider, { children: saltProvider });
  }
  return saltProvider;
}
function SaltProvider({
  enableStyleInjection,
  ...restProps
}) {
  return /* @__PURE__ */ jsxRuntime.jsx(styles.StyleInjectionProvider, { value: enableStyleInjection, children: /* @__PURE__ */ jsxRuntime.jsx(InternalSaltProvider, { ...restProps }) });
}
function SaltProviderNext({
  enableStyleInjection,
  ...restProps
}) {
  return /* @__PURE__ */ jsxRuntime.jsx(styles.StyleInjectionProvider, { value: enableStyleInjection, children: /* @__PURE__ */ jsxRuntime.jsx(InternalSaltProvider, { ...restProps, themeNext: true }) });
}
const UNSTABLE_SaltProviderNext = SaltProviderNext;
const useTheme = () => {
  const { window: _window, ...contextWithoutWindow } = React.useContext(ThemeContext);
  return contextWithoutWindow;
};
function useDensity(density) {
  const densityFromContext = React.useContext(DensityContext);
  return density ?? densityFromContext ?? DEFAULT_DENSITY;
}
const useBreakpoints = () => {
  return React.useContext(BreakpointContext);
};

exports.BreakpointContext = BreakpointContext;
exports.DEFAULT_DENSITY = DEFAULT_DENSITY;
exports.DensityContext = DensityContext;
exports.SaltProvider = SaltProvider;
exports.SaltProviderNext = SaltProviderNext;
exports.ThemeContext = ThemeContext;
exports.UNSTABLE_SaltProviderNext = UNSTABLE_SaltProviderNext;
exports.useBreakpoints = useBreakpoints;
exports.useDensity = useDensity;
exports.useTheme = useTheme;
