// @salt-ds/core/dist-cjs/pill/PillGroup.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
require('../form-field-context/FormFieldContext.js');
var useFormFieldProps = require('../form-field-context/useFormFieldProps.js');
var makePrefixer = require('../utils/makePrefixer.js');
var useControlled = require('../utils/useControlled.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var PillGroup$1 = require('./PillGroup.css.js');
var PillGroupContext = require('./PillGroupContext.js');

const withBaseName = makePrefixer.makePrefixer("saltPillGroup");
const PillGroup = React.forwardRef(
  function PillGroup2(props, ref) {
    const {
      "aria-labelledby": ariaLabelledBy,
      "aria-describedby": ariaDescribedBy,
      disabled: disabledProp,
      children,
      selected: selectedProp,
      defaultSelected,
      onSelectionChange,
      className,
      selectionVariant = "none",
      ...rest
    } = props;
    const targetWindow = window.useWindow();
    styles.useComponentCssInjection({
      testId: "salt-pill-group",
      css: PillGroup$1,
      window: targetWindow
    });
    const { a11yProps: formFieldA11yProps, disabled: formFieldDisabled } = useFormFieldProps.useFormFieldProps();
    const disabled = formFieldDisabled || disabledProp;
    const [selected, setSelected] = useControlled.useControlled({
      controlled: selectedProp,
      default: defaultSelected || [],
      name: "PillGroup",
      state: "selected"
    });
    const select = React.useCallback(
      (event, newValue) => {
        if (disabled) {
          return;
        }
        const newSelected = selected.includes(newValue) ? selected.filter((item) => item !== newValue) : selected.concat(newValue);
        setSelected(newSelected);
        onSelectionChange == null ? void 0 : onSelectionChange(event, newSelected);
      },
      [disabled, selected, onSelectionChange]
    );
    const context = React.useMemo(
      () => ({
        disabled,
        select,
        selected,
        selectionVariant
      }),
      [disabled, select, selected, selectionVariant]
    );
    return /* @__PURE__ */ jsxRuntime.jsx(PillGroupContext.PillGroupContext.Provider, { value: context, children: /* @__PURE__ */ jsxRuntime.jsx(
      "fieldset",
      {
        "aria-labelledby": clsx(formFieldA11yProps == null ? void 0 : formFieldA11yProps["aria-labelledby"], ariaLabelledBy) || void 0,
        "aria-describedby": clsx(formFieldA11yProps == null ? void 0 : formFieldA11yProps["aria-describedby"], ariaDescribedBy) || void 0,
        className: clsx(withBaseName(), className),
        ref,
        ...rest,
        children
      }
    ) });
  }
);

exports.PillGroup = PillGroup;
