// @salt-ds/core/dist-cjs/pill/Pill.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
require('../button/Button.js');
var useButton = require('../button/useButton.js');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var Pill$1 = require('./Pill.css.js');
var PillCheckIcon = require('./PillCheckIcon.js');
var PillGroupContext = require('./PillGroupContext.js');

const withBaseName = makePrefixer.makePrefixer("saltPill");
const Pill = React.forwardRef(function Pill2({
  children,
  className,
  disabled: disabledProp,
  onKeyUp,
  onKeyDown,
  onClick,
  onBlur,
  onPointerDown,
  value,
  ...rest
}, ref) {
  const [pressActive, setPressActive] = React.useState(false);
  const [spaceActive, setSpaceActive] = React.useState(false);
  const pillGroupContext = PillGroupContext.usePillGroup();
  const targetWindow = window.useWindow();
  styles.useComponentCssInjection({
    testId: "salt-pill",
    css: Pill$1,
    window: targetWindow
  });
  const insideSelectableGroup = pillGroupContext.selectionVariant === "multiple";
  const handleClick = (event) => {
    if (insideSelectableGroup && value) {
      pillGroupContext.select(event, value);
    }
    onClick == null ? void 0 : onClick(event);
  };
  const handlePointerDown = (event) => {
    onPointerDown == null ? void 0 : onPointerDown(event);
    if (disabledProp) return;
    setPressActive(true);
  };
  const handleKeyDown = (event) => {
    onKeyDown == null ? void 0 : onKeyDown(event);
    if (event.key === "Enter" && insideSelectableGroup) {
      event.preventDefault();
      return;
    }
    if (event.key === " " && insideSelectableGroup) {
      setSpaceActive(true);
    }
  };
  const handleKeyUp = (event) => {
    onKeyUp == null ? void 0 : onKeyUp(event);
    if (insideSelectableGroup && event.key === " ") {
      setSpaceActive(false);
    }
  };
  React.useEffect(() => {
    if (!targetWindow) return;
    const clear = () => setPressActive(false);
    targetWindow.addEventListener("pointerup", clear);
    targetWindow.addEventListener("pointercancel", clear);
    return () => {
      targetWindow.removeEventListener("pointerup", clear);
      targetWindow.removeEventListener("pointercancel", clear);
    };
  }, [targetWindow]);
  const { buttonProps, active } = useButton.useButton({
    disabled: disabledProp,
    onKeyUp: handleKeyUp,
    onKeyDown: handleKeyDown,
    onClick: handleClick,
    onBlur
  });
  const {
    tabIndex: _tabIndex,
    disabled: buttonDisabled,
    ...restButtonProps
  } = buttonProps;
  const selected = !!value && insideSelectableGroup && pillGroupContext.selected.includes(value);
  const disabled = pillGroupContext.disabled || buttonDisabled;
  const combinedActive = insideSelectableGroup ? pressActive || spaceActive : pressActive || active;
  const groupProps = insideSelectableGroup ? {
    "aria-checked": selected,
    role: "checkbox",
    value
  } : {};
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "button",
    {
      "data-testid": "pill",
      ref,
      className: clsx.clsx(
        withBaseName(),
        withBaseName("clickable"),
        { [withBaseName("active")]: combinedActive },
        className
      ),
      type: "button",
      onPointerDown: handlePointerDown,
      disabled,
      ...restButtonProps,
      ...groupProps,
      ...rest,
      children: [
        insideSelectableGroup && /* @__PURE__ */ jsxRuntime.jsx(PillCheckIcon.PillCheckIcon, { checked: selected, active: combinedActive }),
        children
      ]
    }
  );
});

exports.Pill = Pill;
