// @salt-ds/core/dist-cjs/status-adornment/StatusAdornment.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
var SemanticIconProvider = require('../semantic-icon-provider/SemanticIconProvider.js');
var capitalize = require('../utils/capitalize.js');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var StatusAdornment$1 = require('./StatusAdornment.css.js');

const withBaseName = makePrefixer.makePrefixer("saltStatusAdornment");
const StatusAdornment = React.forwardRef(
  function StatusAdornment2({ className, status, ...restProps }, ref) {
    const icons = SemanticIconProvider.useIcon();
    const titleCaseStatus = capitalize.capitalize(status);
    const iconKey = `${titleCaseStatus}StatusAdornment`;
    const AdornmentComponent = icons[iconKey];
    const ariaLabel = status;
    const targetWindow = window.useWindow();
    styles.useComponentCssInjection({
      testId: "salt-status-adornment",
      css: StatusAdornment$1,
      window: targetWindow
    });
    if (AdornmentComponent === void 0) {
      return null;
    }
    return /* @__PURE__ */ jsxRuntime.jsx(
      AdornmentComponent,
      {
        className: clsx.clsx(withBaseName(), withBaseName(status), className),
        "aria-label": ariaLabel,
        ...restProps,
        ref
      }
    );
  }
);

exports.StatusAdornment = StatusAdornment;
