// @salt-ds/core/dist-es/combo-box/ComboBox.js
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import { offset, size, flip, useInteractions, useDismiss, useFocus, useClick } from '@floating-ui/react';
import { useClassNameInjection } from '@salt-ds/styles';
import { clsx } from 'clsx';
import { forwardRef, useRef, Children, useEffect } from 'react';
import { Button } from '../button/Button.js';
import '../form-field-context/FormFieldContext.js';
import { useFormFieldProps } from '../form-field-context/useFormFieldProps.js';
import { ListControlContext } from '../list-control/ListControlContext.js';
import { defaultValueToString } from '../list-control/ListControlState.js';
import { OptionList } from '../option/OptionList.js';
import { PillInput } from '../pill-input/PillInput.js';
import { useIcon } from '../semantic-icon-provider/SemanticIconProvider.js';
import { makePrefixer } from '../utils/makePrefixer.js';
import { useIsomorphicLayoutEffect } from '../utils/useIsomorphicLayoutEffect.js';
import { useFloatingUI } from '../utils/useFloatingUI/useFloatingUI.js';
import { useForkRef } from '../utils/useForkRef.js';
import { useId } from '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { useComboBox } from './useComboBox.js';

const withBaseName = makePrefixer("saltComboBox");
const ComboBox = forwardRef(function ComboBox2(props, ref) {
  var _a, _b;
  const { className, props: finalProps } = useClassNameInjection(
    "saltComboBox",
    props
  );
  const {
    children,
    disabled: disabledProp,
    endAdornment: endAdornmentProp,
    readOnly: readOnlyProp,
    multiselect,
    selectOnTab = !multiselect,
    onSelectionChange,
    selected,
    defaultSelected,
    defaultOpen,
    onOpenChange,
    onChange,
    open,
    inputRef: inputRefProp,
    inputProps: inputPropsProp,
    variant = "primary",
    onKeyDown,
    onFocus,
    onBlur,
    value,
    defaultValue,
    valueToString = defaultValueToString,
    truncate,
    bordered = false,
    OverlayProps,
    ...rest
  } = finalProps;
  const { CollapseIcon, ExpandIcon } = useIcon();
  const {
    a11yProps: { "aria-labelledby": formFieldLabelledBy } = {},
    disabled: formFieldDisabled,
    readOnly: formFieldReadOnly
  } = useFormFieldProps();
  const disabled = Boolean(disabledProp) || formFieldDisabled;
  const readOnly = Boolean(readOnlyProp) || formFieldReadOnly;
  const inputRef = useRef(null);
  const handleInputRef = useForkRef(inputRef, inputRefProp);
  const shouldAutoSelectRef = useRef(false);
  const listControl = useComboBox({
    open,
    defaultOpen,
    onOpenChange,
    multiselect,
    defaultSelected,
    selected,
    onSelectionChange,
    value,
    defaultValue,
    disabled,
    readOnly,
    valueToString
  });
  const {
    activeState,
    setActive,
    openState,
    setOpen,
    openKey,
    getOptionsMatching,
    getFirstOption,
    getLastOption,
    getOptionAfter,
    getOptionBefore,
    getOptionPageAbove,
    getOptionPageBelow,
    selectedState,
    select,
    clear,
    focusVisibleState,
    setFocusVisibleState,
    focusedState,
    setFocusedState,
    listRef,
    setListRef,
    valueState,
    setValueState,
    removePill
  } = listControl;
  const handleOpenChange = (newOpen, _event, reason) => {
    var _a2;
    const focusNotBlur = reason === "focus" && newOpen;
    if (reason === "focus") {
      setFocusedState(newOpen);
    }
    if (reason === "focus" && !newOpen) {
      setFocusVisibleState(false);
    }
    if (readOnly || focusNotBlur) return;
    setOpen(newOpen);
    if (newOpen) {
      (_a2 = inputRef.current) == null ? void 0 : _a2.focus();
    }
  };
  const hasValidChildren = Children.toArray(children).filter(Boolean).length > 0;
  const { x, y, strategy, elements, floating, reference, context } = useFloatingUI({
    open: openState && !readOnly && hasValidChildren,
    onOpenChange: handleOpenChange,
    placement: "bottom-start",
    strategy: "fixed",
    middleware: [
      offset(1),
      size({
        apply({ rects, elements: elements2, availableHeight }) {
          elements2.floating.style.setProperty(
            "--overlay-minWidth",
            `${rects.reference.width}px`
          );
          elements2.floating.style.setProperty(
            "--overlay-maxHeight",
            `max(calc((var(--salt-size-base) + var(--salt-spacing-100)) * 5), calc(${availableHeight}px - var(--salt-spacing-100)))`
          );
        }
      }),
      flip({ fallbackStrategy: "initialPlacement" })
    ]
  });
  const { getReferenceProps, getFloatingProps } = useInteractions([
    useDismiss(context),
    useFocus(context),
    useClick(context, { keyboardHandlers: false, toggle: false })
  ]);
  const handleRef = useForkRef(reference, ref);
  const handleButtonClick = (event) => {
    if (!readOnly) {
      event.stopPropagation();
      setFocusVisibleState(false);
      setOpen(!openState, "manual");
    }
  };
  const handleButtonFocus = () => {
    var _a2;
    (_a2 = inputRef.current) == null ? void 0 : _a2.focus();
  };
  const handleKeyDown = (event) => {
    onKeyDown == null ? void 0 : onKeyDown(event);
    if (!event.repeat) {
      shouldAutoSelectRef.current = false;
    }
    if (readOnly) {
      return;
    }
    if (!openState) {
      if (event.key === "ArrowDown" || event.key === "ArrowUp") {
        setOpen(true, void 0, event.key);
        return;
      }
    }
    const activeOption = activeState;
    let newActive;
    switch (event.key) {
      case "ArrowDown":
        newActive = activeOption ? getOptionAfter(activeOption) : getFirstOption();
        break;
      case "ArrowUp":
        newActive = activeOption ? getOptionBefore(activeOption) : getLastOption();
        break;
      case "Home":
        newActive = getFirstOption();
        break;
      case "End":
        newActive = getLastOption();
        break;
      case "PageUp":
        if (activeOption) {
          newActive = getOptionPageAbove(activeOption);
        } else {
          const lastOption = getLastOption();
          if (lastOption) {
            newActive = getOptionPageAbove(lastOption == null ? void 0 : lastOption.data);
          }
        }
        break;
      case "PageDown":
        if (activeOption) {
          newActive = getOptionPageBelow(activeOption);
        } else {
          const firstOption = getFirstOption();
          if (firstOption) {
            newActive = getOptionPageBelow(firstOption.data);
          }
        }
        break;
      case "Enter":
        if (openState && (activeState == null ? void 0 : activeState.disabled)) {
          event.preventDefault();
          return;
        }
        if (!openState || !activeState) {
          return;
        }
        select(event, activeState);
        if (!multiselect) {
          event.preventDefault();
        }
        break;
      case "Tab":
        if (openState && selectOnTab && activeState && !(activeState == null ? void 0 : activeState.disabled) && !selectedState.includes(activeState == null ? void 0 : activeState.value)) {
          select(event, activeState);
        }
        break;
    }
    if (newActive) {
      setFocusVisibleState(true);
    }
    if (newActive && newActive.data.id !== (activeState == null ? void 0 : activeState.id)) {
      event.preventDefault();
      setActive(newActive.data);
    }
  };
  const handleFocus = (event) => {
    if (event.currentTarget === inputRef.current) {
      setFocusedState(true);
    } else {
      setActive(void 0);
    }
    onFocus == null ? void 0 : onFocus(event);
  };
  const handleBlur = (event) => {
    event.persist();
    if (!listRef.current || !listRef.current.contains(event.relatedTarget)) {
      onBlur == null ? void 0 : onBlur(event);
      shouldAutoSelectRef.current = false;
    }
  };
  const handleChange = (event) => {
    const value2 = event.target.value;
    if (!openState) {
      setOpen(true, "input");
    }
    if (value2 === "" && !multiselect) {
      clear(event);
    }
    setValueState(value2);
    if (value2 === "") {
      setActive(void 0);
    } else {
      shouldAutoSelectRef.current = true;
    }
    onChange == null ? void 0 : onChange(event);
  };
  const handlePillRemove = (event, index) => {
    event.stopPropagation();
    const removed = selectedState[index];
    removePill(event, removed);
  };
  const handleListMouseOver = () => {
    setFocusVisibleState(false);
  };
  const handleListMouseLeave = () => {
    setActive(void 0);
  };
  const handleFocusInput = () => {
    var _a2;
    (_a2 = inputRef.current) == null ? void 0 : _a2.focus();
  };
  useIsomorphicLayoutEffect(() => {
    if (value) {
      shouldAutoSelectRef.current = true;
    }
  }, [value]);
  useEffect(() => {
    if (openState && value) {
      queueMicrotask(() => {
        const newOption = getFirstOption();
        if (newOption && shouldAutoSelectRef.current) {
          setActive(newOption.data);
        }
      });
    }
  }, [value, setActive, openState, getFirstOption]);
  useEffect(() => {
    let newActive;
    if (!openState) {
      setActive(void 0);
      return;
    }
    if (selectedState.length > 0) {
      newActive = getOptionsMatching(
        (option) => option.value === selectedState[0]
      ).pop();
    }
    if (!newActive) {
      if (openKey.current === "ArrowDown") {
        newActive = getFirstOption();
        setFocusVisibleState(true);
      } else if (openKey.current === "ArrowUp") {
        newActive = getLastOption();
        setFocusVisibleState(true);
      }
    }
    if (newActive) {
      setActive(newActive == null ? void 0 : newActive.data);
    }
  }, [openState]);
  const buttonId = useId();
  const listId = useId();
  const handleListRef = useForkRef(setListRef, floating);
  const showOptionsButton = /* @__PURE__ */ jsx(
    Button,
    {
      "aria-labelledby": clsx(buttonId, formFieldLabelledBy),
      "aria-label": "Show options",
      "aria-expanded": openState,
      "aria-controls": openState ? listId : void 0,
      "aria-haspopup": "listbox",
      disabled,
      appearance: "transparent",
      onClick: handleButtonClick,
      onFocus: handleButtonFocus,
      tabIndex: -1,
      children: openState ? /* @__PURE__ */ jsx(CollapseIcon, { "aria-hidden": true }) : /* @__PURE__ */ jsx(ExpandIcon, { "aria-hidden": true })
    }
  );
  const endAdornment = !readOnly && hasValidChildren ? /* @__PURE__ */ jsxs(Fragment, { children: [
    endAdornmentProp,
    showOptionsButton
  ] }) : endAdornmentProp;
  return /* @__PURE__ */ jsxs(ListControlContext.Provider, { value: listControl, children: [
    /* @__PURE__ */ jsx(
      PillInput,
      {
        tabIndex: !disabled ? -1 : void 0,
        className: clsx(
          withBaseName(),
          {
            [withBaseName("focused")]: focusedState,
            [withBaseName("focusVisible")]: focusVisibleState
          },
          className
        ),
        endAdornment,
        onChange: handleChange,
        role: readOnly ? "textbox" : "combobox",
        disabled,
        readOnly,
        inputProps: {
          "aria-readonly": readOnly ? "true" : void 0,
          "aria-expanded": !readOnly ? openState : void 0,
          "aria-controls": openState && !readOnly ? listId : void 0,
          onKeyDown: handleKeyDown,
          ...inputPropsProp
        },
        "aria-activedescendant": activeState == null ? void 0 : activeState.id,
        variant,
        inputRef: handleInputRef,
        value: valueState,
        ref: handleRef,
        bordered,
        ...getReferenceProps({
          onBlur: handleBlur,
          onFocus: handleFocus,
          ...rest
        }),
        pills: multiselect ? selectedState.map((item) => valueToString(item)) : [],
        truncate: truncate && !focusedState && !openState,
        onPillRemove: handlePillRemove,
        hidePillClose: !focusedState || readOnly,
        emptyReadOnlyMarker: readOnly && selectedState.length > 0 ? "" : void 0
      }
    ),
    /* @__PURE__ */ jsx(
      OptionList,
      {
        "aria-multiselectable": multiselect,
        open: (openState || focusedState) && !readOnly && hasValidChildren,
        collapsed: !openState,
        ref: handleListRef,
        id: listId,
        tabIndex: -1,
        ...getFloatingProps({
          onMouseOver: handleListMouseOver,
          onFocus: handleFocusInput,
          onClick: handleFocusInput,
          onMouseLeave: handleListMouseLeave,
          ...OverlayProps
        }),
        left: x ?? 0,
        top: y ?? 0,
        position: strategy,
        width: (_a = elements.floating) == null ? void 0 : _a.offsetWidth,
        height: (_b = elements.floating) == null ? void 0 : _b.offsetHeight,
        children
      }
    )
  ] });
});

export { ComboBox };
