// @salt-ds/core/dist-es/pill/PillGroup.js
import { jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import clsx from 'clsx';
import { forwardRef, useCallback, useMemo } from 'react';
import '../form-field-context/FormFieldContext.js';
import { useFormFieldProps } from '../form-field-context/useFormFieldProps.js';
import { makePrefixer } from '../utils/makePrefixer.js';
import { useControlled } from '../utils/useControlled.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import css_248z from './PillGroup.css.js';
import { PillGroupContext } from './PillGroupContext.js';

const withBaseName = makePrefixer("saltPillGroup");
const PillGroup = forwardRef(
  function PillGroup2(props, ref) {
    const {
      "aria-labelledby": ariaLabelledBy,
      "aria-describedby": ariaDescribedBy,
      disabled: disabledProp,
      children,
      selected: selectedProp,
      defaultSelected,
      onSelectionChange,
      className,
      selectionVariant = "none",
      ...rest
    } = props;
    const targetWindow = useWindow();
    useComponentCssInjection({
      testId: "salt-pill-group",
      css: css_248z,
      window: targetWindow
    });
    const { a11yProps: formFieldA11yProps, disabled: formFieldDisabled } = useFormFieldProps();
    const disabled = formFieldDisabled || disabledProp;
    const [selected, setSelected] = useControlled({
      controlled: selectedProp,
      default: defaultSelected || [],
      name: "PillGroup",
      state: "selected"
    });
    const select = useCallback(
      (event, newValue) => {
        if (disabled) {
          return;
        }
        const newSelected = selected.includes(newValue) ? selected.filter((item) => item !== newValue) : selected.concat(newValue);
        setSelected(newSelected);
        onSelectionChange == null ? void 0 : onSelectionChange(event, newSelected);
      },
      [disabled, selected, onSelectionChange]
    );
    const context = useMemo(
      () => ({
        disabled,
        select,
        selected,
        selectionVariant
      }),
      [disabled, select, selected, selectionVariant]
    );
    return /* @__PURE__ */ jsx(PillGroupContext.Provider, { value: context, children: /* @__PURE__ */ jsx(
      "fieldset",
      {
        "aria-labelledby": clsx(formFieldA11yProps == null ? void 0 : formFieldA11yProps["aria-labelledby"], ariaLabelledBy) || void 0,
        "aria-describedby": clsx(formFieldA11yProps == null ? void 0 : formFieldA11yProps["aria-describedby"], ariaDescribedBy) || void 0,
        className: clsx(withBaseName(), className),
        ref,
        ...rest,
        children
      }
    ) });
  }
);

export { PillGroup };
