// @salt-ds/core/dist-es/pill/Pill.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef, useState, useEffect } from 'react';
import '../button/Button.js';
import { useButton } from '../button/useButton.js';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import css_248z from './Pill.css.js';
import { PillCheckIcon } from './PillCheckIcon.js';
import { usePillGroup } from './PillGroupContext.js';

const withBaseName = makePrefixer("saltPill");
const Pill = forwardRef(function Pill2({
  children,
  className,
  disabled: disabledProp,
  onKeyUp,
  onKeyDown,
  onClick,
  onBlur,
  onPointerDown,
  value,
  ...rest
}, ref) {
  const [pressActive, setPressActive] = useState(false);
  const [spaceActive, setSpaceActive] = useState(false);
  const pillGroupContext = usePillGroup();
  const targetWindow = useWindow();
  useComponentCssInjection({
    testId: "salt-pill",
    css: css_248z,
    window: targetWindow
  });
  const insideSelectableGroup = pillGroupContext.selectionVariant === "multiple";
  const handleClick = (event) => {
    if (insideSelectableGroup && value) {
      pillGroupContext.select(event, value);
    }
    onClick == null ? void 0 : onClick(event);
  };
  const handlePointerDown = (event) => {
    onPointerDown == null ? void 0 : onPointerDown(event);
    if (disabledProp) return;
    setPressActive(true);
  };
  const handleKeyDown = (event) => {
    onKeyDown == null ? void 0 : onKeyDown(event);
    if (event.key === "Enter" && insideSelectableGroup) {
      event.preventDefault();
      return;
    }
    if (event.key === " " && insideSelectableGroup) {
      setSpaceActive(true);
    }
  };
  const handleKeyUp = (event) => {
    onKeyUp == null ? void 0 : onKeyUp(event);
    if (insideSelectableGroup && event.key === " ") {
      setSpaceActive(false);
    }
  };
  useEffect(() => {
    if (!targetWindow) return;
    const clear = () => setPressActive(false);
    targetWindow.addEventListener("pointerup", clear);
    targetWindow.addEventListener("pointercancel", clear);
    return () => {
      targetWindow.removeEventListener("pointerup", clear);
      targetWindow.removeEventListener("pointercancel", clear);
    };
  }, [targetWindow]);
  const { buttonProps, active } = useButton({
    disabled: disabledProp,
    onKeyUp: handleKeyUp,
    onKeyDown: handleKeyDown,
    onClick: handleClick,
    onBlur
  });
  const {
    tabIndex: _tabIndex,
    disabled: buttonDisabled,
    ...restButtonProps
  } = buttonProps;
  const selected = !!value && insideSelectableGroup && pillGroupContext.selected.includes(value);
  const disabled = pillGroupContext.disabled || buttonDisabled;
  const combinedActive = insideSelectableGroup ? pressActive || spaceActive : pressActive || active;
  const groupProps = insideSelectableGroup ? {
    "aria-checked": selected,
    role: "checkbox",
    value
  } : {};
  return /* @__PURE__ */ jsxs(
    "button",
    {
      "data-testid": "pill",
      ref,
      className: clsx(
        withBaseName(),
        withBaseName("clickable"),
        { [withBaseName("active")]: combinedActive },
        className
      ),
      type: "button",
      onPointerDown: handlePointerDown,
      disabled,
      ...restButtonProps,
      ...groupProps,
      ...rest,
      children: [
        insideSelectableGroup && /* @__PURE__ */ jsx(PillCheckIcon, { checked: selected, active: combinedActive }),
        children
      ]
    }
  );
});

export { Pill };
