// @salt-ds/core/dist-es/salt-provider/SaltProvider.js
import { jsx } from 'react/jsx-runtime';
import { StyleInjectionProvider, useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { createContext, useContext, useMemo } from 'react';
import '../aria-announcer/AriaAnnouncerContext.js';
import { AriaAnnouncerProvider } from '../aria-announcer/AriaAnnouncerProvider.js';
import { useMatchedBreakpoints, BreakpointProvider } from '../breakpoints/BreakpointProvider.js';
import { DEFAULT_BREAKPOINTS } from '../breakpoints/Breakpoints.js';
import { ViewportProvider } from '../viewport/ViewportProvider.js';
import { ProviderContext } from './ProviderContext.js';
import css_248z from './SaltProvider.css.js';
import { ThemeApplicator } from './ThemeApplicator.js';

const DEFAULT_DENSITY = "medium";
const DEFAULT_THEME_NAME = "salt-theme";
const DEFAULT_THEME_NAME_NEXT = "salt-theme-next";
const DEFAULT_MODE = "light";
const DEFAULT_CORNER = "sharp";
const DEFAULT_HEADING_FONT = "Open Sans";
const DEFAULT_ACCENT = "blue";
const DEFAULT_ACTION_FONT = "Open Sans";
const DensityContext = createContext(DEFAULT_DENSITY);
const ThemeContext = createContext({
  theme: "",
  mode: DEFAULT_MODE,
  themeNext: false,
  corner: DEFAULT_CORNER,
  UNSTABLE_corner: DEFAULT_CORNER,
  headingFont: DEFAULT_HEADING_FONT,
  UNSTABLE_headingFont: DEFAULT_HEADING_FONT,
  accent: DEFAULT_ACCENT,
  UNSTABLE_accent: DEFAULT_ACCENT,
  actionFont: DEFAULT_ACTION_FONT,
  UNSTABLE_actionFont: DEFAULT_ACTION_FONT
});
const BreakpointContext = createContext(DEFAULT_BREAKPOINTS);
function InternalSaltProvider({
  applyClassesTo: applyClassesToProp,
  children,
  density: densityProp,
  theme: themeProp,
  mode: modeProp,
  breakpoints: breakpointsProp,
  themeNext,
  corner: cornerProp,
  headingFont: headingFontProp,
  accent: accentProp,
  actionFont: actionFontProp
}) {
  const prevProvider = useContext(ProviderContext);
  const inheritedDensity = useContext(DensityContext);
  const {
    theme: inheritedTheme,
    mode: inheritedMode,
    corner: inheritedCorner,
    headingFont: inheritedHeadingFont,
    accent: inheritedAccent,
    actionFont: inheritedActionFont
  } = useContext(ThemeContext);
  const isRootProvider = prevProvider === null;
  const density = densityProp ?? inheritedDensity ?? DEFAULT_DENSITY;
  const inheritedThemeName = inheritedTheme === "" ? DEFAULT_THEME_NAME : inheritedTheme;
  const themeName = themeProp ?? inheritedThemeName;
  const mode = modeProp ?? inheritedMode;
  const breakpoints = breakpointsProp ?? DEFAULT_BREAKPOINTS;
  const corner = cornerProp ?? inheritedCorner ?? DEFAULT_CORNER;
  const headingFont = headingFontProp ?? inheritedHeadingFont ?? DEFAULT_HEADING_FONT;
  const accent = accentProp ?? inheritedAccent ?? DEFAULT_ACCENT;
  const actionFont = actionFontProp ?? inheritedActionFont ?? DEFAULT_ACTION_FONT;
  const targetWindow = useWindow();
  useComponentCssInjection({
    testId: "salt-provider",
    css: css_248z,
    window: targetWindow
  });
  const themeContextValue = useMemo(
    () => ({
      theme: themeName,
      mode,
      window: targetWindow,
      themeNext: Boolean(themeNext),
      corner,
      headingFont,
      accent,
      actionFont,
      // Backward compatibility
      UNSTABLE_corner: corner,
      UNSTABLE_headingFont: headingFont,
      UNSTABLE_accent: accent,
      UNSTABLE_actionFont: actionFont
    }),
    [
      themeName,
      mode,
      targetWindow,
      themeNext,
      corner,
      headingFont,
      accent,
      actionFont
    ]
  );
  const matchedBreakpoints = useMatchedBreakpoints(breakpoints);
  const saltProvider = /* @__PURE__ */ jsx(ProviderContext.Provider, { value: { targetWindow }, children: /* @__PURE__ */ jsx(DensityContext.Provider, { value: density, children: /* @__PURE__ */ jsx(ThemeContext.Provider, { value: themeContextValue, children: /* @__PURE__ */ jsx(BreakpointProvider, { matchedBreakpoints, children: /* @__PURE__ */ jsx(BreakpointContext.Provider, { value: breakpoints, children: /* @__PURE__ */ jsx(ViewportProvider, { children: /* @__PURE__ */ jsx(
    ThemeApplicator,
    {
      applyClassesTo: applyClassesToProp ?? (isRootProvider ? "root" : "scope"),
      density,
      theme: clsx(
        DEFAULT_THEME_NAME,
        { [DEFAULT_THEME_NAME_NEXT]: themeNext },
        themeProp
      ),
      mode,
      accent: themeNext ? accent : void 0,
      actionFont: themeNext ? actionFont : void 0,
      headingFont: themeNext ? headingFont : void 0,
      corner: themeNext ? corner : void 0,
      children
    }
  ) }) }) }) }) }) });
  if (isRootProvider) {
    return /* @__PURE__ */ jsx(AriaAnnouncerProvider, { children: saltProvider });
  }
  return saltProvider;
}
function SaltProvider({
  enableStyleInjection,
  ...restProps
}) {
  return /* @__PURE__ */ jsx(StyleInjectionProvider, { value: enableStyleInjection, children: /* @__PURE__ */ jsx(InternalSaltProvider, { ...restProps }) });
}
function SaltProviderNext({
  enableStyleInjection,
  ...restProps
}) {
  return /* @__PURE__ */ jsx(StyleInjectionProvider, { value: enableStyleInjection, children: /* @__PURE__ */ jsx(InternalSaltProvider, { ...restProps, themeNext: true }) });
}
const UNSTABLE_SaltProviderNext = SaltProviderNext;
const useTheme = () => {
  const { window: _window, ...contextWithoutWindow } = useContext(ThemeContext);
  return contextWithoutWindow;
};
function useDensity(density) {
  const densityFromContext = useContext(DensityContext);
  return density ?? densityFromContext ?? DEFAULT_DENSITY;
}
const useBreakpoints = () => {
  return useContext(BreakpointContext);
};

export { BreakpointContext, DEFAULT_DENSITY, DensityContext, SaltProvider, SaltProviderNext, ThemeContext, UNSTABLE_SaltProviderNext, useBreakpoints, useDensity, useTheme };
