// @salt-ds/core/dist-es/status-adornment/StatusAdornment.js
import { jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef } from 'react';
import { useIcon } from '../semantic-icon-provider/SemanticIconProvider.js';
import { capitalize } from '../utils/capitalize.js';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import css_248z from './StatusAdornment.css.js';

const withBaseName = makePrefixer("saltStatusAdornment");
const StatusAdornment = forwardRef(
  function StatusAdornment2({ className, status, ...restProps }, ref) {
    const icons = useIcon();
    const titleCaseStatus = capitalize(status);
    const iconKey = `${titleCaseStatus}StatusAdornment`;
    const AdornmentComponent = icons[iconKey];
    const ariaLabel = status;
    const targetWindow = useWindow();
    useComponentCssInjection({
      testId: "salt-status-adornment",
      css: css_248z,
      window: targetWindow
    });
    if (AdornmentComponent === void 0) {
      return null;
    }
    return /* @__PURE__ */ jsx(
      AdornmentComponent,
      {
        className: clsx(withBaseName(), withBaseName(status), className),
        "aria-label": ariaLabel,
        ...restProps,
        ref
      }
    );
  }
);

export { StatusAdornment };
