// @salt-ds/core/dist-es/table/Table.js
import { jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef } from 'react';
import { makePrefixer } from '../utils/makePrefixer.js';
import { useIsomorphicLayoutEffect } from '../utils/useIsomorphicLayoutEffect.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import { useId } from '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import css_248z from './Table.css.js';
import { useTable } from './TableContext.js';

const withTableBaseName = makePrefixer("saltTable");
const Table = forwardRef(
  function Table2(props, ref) {
    const targetWindow = useWindow();
    useComponentCssInjection({
      testId: "salt-table",
      css: css_248z,
      window: targetWindow
    });
    const {
      children,
      className,
      variant = "primary",
      divider = "tertiary",
      zebra = false,
      "aria-labelledby": ariaLabelledBy,
      id,
      ...rest
    } = props;
    const tableId = useId(id);
    const { setId, setLabelledBy } = useTable();
    const labelledBy = ariaLabelledBy ?? tableId;
    useIsomorphicLayoutEffect(() => {
      if (tableId) {
        setId(tableId);
      }
      if (labelledBy) {
        setLabelledBy(labelledBy);
      }
    }, [tableId, labelledBy, setId, setLabelledBy]);
    return /* @__PURE__ */ jsx(
      "table",
      {
        id: tableId,
        className: clsx(
          withTableBaseName(),
          {
            [withTableBaseName(variant)]: variant,
            [withTableBaseName("zebra")]: zebra,
            [withTableBaseName(`divider-${divider}`)]: divider
          },
          className
        ),
        "aria-labelledby": labelledBy,
        ref,
        ...rest,
        children
      }
    );
  }
);

export { Table, withTableBaseName };
