// @salt-ds/core/dist-es/table/TableContainer.js
import { jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef, useState, useRef, useCallback, useMemo } from 'react';
import { useIsomorphicLayoutEffect } from '../utils/useIsomorphicLayoutEffect.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import { useForkRef } from '../utils/useForkRef.js';
import '../utils/useId.js';
import { useResizeObserver } from '../utils/useResizeObserver.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { withTableBaseName } from './Table.js';
import css_248z from './Table.css.js';
import { TableContext } from './TableContext.js';

const TableContainer = forwardRef(
  function TableContainer2(props, ref) {
    const [isOverflowing, setIsOverflowing] = useState(false);
    const [tableId, setTableId] = useState();
    const [labelledBy, setLabelledBy] = useState();
    const targetWindow = useWindow();
    useComponentCssInjection({
      testId: "salt-table-container",
      css: css_248z,
      window: targetWindow
    });
    const {
      children,
      className,
      // @ts-expect-error: "aria-labelledby" is omitted to prevent accidental misuse,
      // but we still want to forward it for advanced accessible labeling scenarios.
      "aria-labelledby": ariaLabelledBy,
      // @ts-expect-error: Allow passing aria-label even though it's omitted from HTMLAttributes
      // Same reasoning as above: we forward aria-label for accessibility purposes.
      "aria-label": ariaLabel,
      // @ts-expect-error: Allow passing role even though it's omitted from HTMLAttributes
      // Same reasoning as above: we forward role for accessibility purposes.
      role,
      tabIndex,
      ...rest
    } = props;
    const scrollRef = useRef(null);
    const handleRef = useForkRef(ref, scrollRef);
    const checkOverflow = useCallback(() => {
      const element = scrollRef.current;
      if (!element) return;
      const overflowing = element.scrollHeight > element.clientHeight || element.scrollWidth > element.clientWidth;
      setIsOverflowing(overflowing);
    }, []);
    useResizeObserver({ ref: scrollRef, onResize: checkOverflow });
    useIsomorphicLayoutEffect(() => {
      checkOverflow();
    }, [checkOverflow]);
    const overflowProps = isOverflowing ? {
      role: role ?? "region",
      tabIndex: tabIndex ?? 0,
      "aria-label": ariaLabel,
      ...ariaLabelledBy === void 0 && ariaLabel === void 0 && {
        "aria-labelledby": labelledBy ?? tableId
      },
      ...ariaLabelledBy !== void 0 && {
        "aria-labelledby": ariaLabelledBy
      }
    } : {
      role,
      tabIndex,
      "aria-labelledby": ariaLabelledBy,
      "aria-label": ariaLabel
    };
    const contextValue = useMemo(
      () => ({ id: tableId, setId: setTableId, labelledBy, setLabelledBy }),
      [tableId, labelledBy]
    );
    return /* @__PURE__ */ jsx(TableContext.Provider, { value: contextValue, children: /* @__PURE__ */ jsx(
      "div",
      {
        ref: handleRef,
        className: clsx(withTableBaseName("container"), className),
        ...overflowProps,
        ...rest,
        children
      }
    ) });
  }
);

export { TableContainer };
