// @salt-ds/core/dist-es/table/TFoot.js
import { jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef } from 'react';
import { withTableBaseName } from './Table.js';
import css_248z from './Table.css.js';

const TFoot = forwardRef(
  function TFoot2({
    children,
    className,
    sticky = false,
    variant,
    divider = "primary",
    ...rest
  }, ref) {
    const targetWindow = useWindow();
    useComponentCssInjection({
      testId: "salt-table-tfoot",
      css: css_248z,
      window: targetWindow
    });
    return /* @__PURE__ */ jsx(
      "tfoot",
      {
        ref,
        className: clsx(
          withTableBaseName("tfoot"),
          { [withTableBaseName(`tfoot-${variant}`)]: variant },
          { [withTableBaseName("tfoot-sticky")]: sticky },
          { [withTableBaseName(`tfoot-divider-${divider}`)]: divider },
          className
        ),
        ...rest,
        children
      }
    );
  }
);

export { TFoot };
