// @salt-ds/core/dist-cjs/list-control/ListControlContext.js
'use strict';

var React = require('react');
var createContext = require('../utils/createContext.js');
require('clsx');
require('react/jsx-runtime');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var ListControlState = require('./ListControlState.js');

const ListControlContext = createContext.createContext(
  "ListControlContext",
  {
    openState: false,
    setOpen() {
      return void 0;
    },
    register() {
      return () => void 0;
    },
    selectedState: [],
    select() {
      return void 0;
    },
    activeState: void 0,
    setActive() {
      return void 0;
    },
    multiselect: false,
    focusVisibleState: false,
    valueToString: ListControlState.defaultValueToString,
    disabled: false,
    listRef: void 0
  }
);
function useListControlContext() {
  return React.useContext(ListControlContext);
}

exports.ListControlContext = ListControlContext;
exports.useListControlContext = useListControlContext;
