// @salt-ds/core/dist-cjs/list-control/ListControlState.js
'use strict';

var React = require('react');
require('clsx');
require('react/jsx-runtime');
var useControlled = require('../utils/useControlled.js');
require('../utils/useFloatingUI/useFloatingUI.js');
var useForkRef = require('../utils/useForkRef.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');

function defaultValueToString(item) {
  return String(item);
}
function useListControl(props) {
  const {
    open: openProp,
    defaultOpen,
    onOpenChange,
    multiselect,
    defaultSelected,
    selected: selectedProp,
    onSelectionChange,
    disabled,
    readOnly,
    valueToString = defaultValueToString
  } = props;
  const listRef = React.useRef(null);
  const [listElement, setListElement] = React.useState(null);
  const setListRef = useForkRef.useForkRef(listRef, setListElement);
  const [focusedState, setFocusedState] = React.useState(false);
  const [focusVisibleState, setFocusVisibleState] = React.useState(false);
  React.useEffect(() => {
    if (disabled && focusedState) {
      setFocusedState(false);
      setFocusVisibleState(false);
    }
  }, [disabled, focusedState]);
  const [activeState, setActiveState] = React.useState(
    void 0
  );
  const setActive = React.useCallback((option) => {
    if (option) {
      setActiveState(option);
    } else {
      setActiveState(void 0);
    }
  }, []);
  const [openState, setOpenState] = useControlled.useControlled({
    controlled: openProp,
    default: Boolean(defaultOpen),
    name: "ListControl",
    state: "open"
  });
  const openKey = React.useRef(void 0);
  const setOpen = (newOpen, reason, key) => {
    if (disabled || readOnly) {
      return;
    }
    setOpenState(newOpen);
    openKey.current = key;
    if (newOpen !== openState) {
      onOpenChange == null ? void 0 : onOpenChange(newOpen, reason);
    }
  };
  const [selectedState, setSelectedState] = useControlled.useControlled({
    controlled: selectedProp,
    default: defaultSelected ?? [],
    name: "ListControl",
    state: "selected"
  });
  const select = (event, option) => {
    if (option.disabled || readOnly || disabled) {
      return;
    }
    let newSelected = [option.value];
    if (multiselect) {
      if (selectedState.includes(option.value)) {
        newSelected = selectedState.filter((item) => item !== option.value);
      } else {
        newSelected = selectedState.concat([option.value]);
      }
    }
    setSelectedState(newSelected);
    onSelectionChange == null ? void 0 : onSelectionChange(event, newSelected);
    if (!multiselect) {
      setOpen(false);
    }
  };
  const clear = (event) => {
    setSelectedState([]);
    if (selectedState.length !== 0) {
      onSelectionChange == null ? void 0 : onSelectionChange(event, []);
    }
  };
  const optionsRef = React.useRef([]);
  const register = React.useCallback(
    (optionValue, element) => {
      const { id } = optionValue;
      const existingIndex = optionsRef.current.findIndex(
        (item) => item.data.id === id
      );
      if (existingIndex !== -1) {
        optionsRef.current.splice(existingIndex, 1);
      }
      optionsRef.current.push({ data: optionValue, element });
      return () => {
        optionsRef.current = optionsRef.current.filter(
          (item) => item.data.id !== id
        );
      };
    },
    []
  );
  React.useEffect(() => {
    const sortOptions = () => {
      optionsRef.current = optionsRef.current.filter((a) => a.element.isConnected).sort(({ element: a }, { element: b }) => {
        if (a === b) return 0;
        const pos = a.compareDocumentPosition(b);
        if (pos & Node.DOCUMENT_POSITION_FOLLOWING) return -1;
        if (pos & Node.DOCUMENT_POSITION_PRECEDING) return 1;
        return 0;
      });
    };
    const mutationObserver = new MutationObserver((mutations) => {
      const optionsChanged = mutations.some(
        (mutation) => Array.from(mutation.addedNodes).some(
          (node) => {
            var _a;
            return node instanceof HTMLElement && ((_a = node.matches) == null ? void 0 : _a.call(node, '[role="option"]'));
          }
        )
      );
      if (optionsChanged) {
        sortOptions();
      }
    });
    if (!listElement) return;
    mutationObserver.observe(listElement, {
      childList: true,
      subtree: true
    });
    sortOptions();
    return () => mutationObserver.disconnect();
  }, [listElement]);
  const getOptionAtIndex = React.useCallback(
    (index) => {
      return optionsRef.current[index];
    },
    []
  );
  const getIndexOfOption = React.useCallback((option) => {
    return optionsRef.current.findIndex(
      (item) => item.data.value === option.value
    );
  }, []);
  const getOptionsMatching = React.useCallback(
    (predicate) => {
      return optionsRef.current.filter((item) => predicate(item.data));
    },
    []
  );
  const getOptionFromSearch = React.useCallback(
    (search, startFrom) => {
      const collator = new Intl.Collator("en", {
        usage: "search",
        sensitivity: "base"
      });
      const startIndex = startFrom ? getIndexOfOption(startFrom) + 1 : 0;
      const searchList = optionsRef.current.map((item) => item.data);
      let matches = searchList.filter(
        (option) => collator.compare(
          valueToString(option.value).substring(0, search.length),
          search
        ) === 0
      );
      if (matches.length === 0) {
        const letters = search.split("");
        const allSameLetter = letters.length > 0 && letters.every((letter) => collator.compare(letter, letters[0]) === 0);
        if (allSameLetter) {
          matches = searchList.filter(
            (option) => collator.compare(
              valueToString(option.value)[0].toLowerCase(),
              letters[0]
            ) === 0
          );
        }
      }
      return matches.find((option) => getIndexOfOption(option) >= startIndex);
    },
    [getIndexOfOption, valueToString]
  );
  const getFirstOption = React.useCallback(() => {
    return getOptionAtIndex(0);
  }, [getOptionAtIndex]);
  const getLastOption = React.useCallback(() => {
    return getOptionAtIndex(optionsRef.current.length - 1);
  }, [getOptionAtIndex]);
  const getOptionBefore = React.useCallback(
    (option) => {
      const index = getIndexOfOption(option);
      return getOptionAtIndex(index - 1);
    },
    [getIndexOfOption, getOptionAtIndex]
  );
  const getOptionAfter = React.useCallback(
    (option) => {
      const index = getIndexOfOption(option);
      return getOptionAtIndex(index + 1);
    },
    [getIndexOfOption, getOptionAtIndex]
  );
  const getOptionPageAbove = React.useCallback(
    (start) => {
      var _a;
      const list = listRef.current;
      let option = optionsRef.current.find((option2) => option2.data === start);
      if (!list || !option) {
        return void 0;
      }
      const containerRect = list.getBoundingClientRect();
      let optionRect = option.element.getBoundingClientRect();
      const listY = containerRect.y - list.scrollTop;
      const pageY = Math.max(
        0,
        optionRect.y - listY + optionRect.height - containerRect.height
      );
      while (option && optionRect && optionRect.y - listY > pageY) {
        option = getOptionBefore(option.data);
        optionRect = (_a = option == null ? void 0 : option.element) == null ? void 0 : _a.getBoundingClientRect();
      }
      return option ?? getFirstOption();
    },
    [getFirstOption, getOptionBefore]
  );
  const getOptionPageBelow = React.useCallback(
    (start) => {
      const list = listRef.current;
      let option = optionsRef.current.find((option2) => option2.data === start);
      if (!list || !option) {
        return void 0;
      }
      const containerRect = list.getBoundingClientRect();
      let optionRect = option.element.getBoundingClientRect();
      const listY = containerRect.y - list.scrollTop;
      const pageY = Math.min(
        list.scrollHeight,
        optionRect.y - listY - optionRect.height + containerRect.height
      );
      while (option && optionRect && optionRect.y - listY < pageY) {
        option = getOptionAfter(option.data);
        optionRect = option == null ? void 0 : option.element.getBoundingClientRect();
      }
      return option ?? getLastOption();
    },
    [getLastOption, getOptionAfter]
  );
  React.useEffect(() => {
    var _a;
    if (listRef.current) {
      const activeElement = (_a = optionsRef.current.find(
        (option) => option.data === activeState
      )) == null ? void 0 : _a.element;
      if (!activeElement) {
        return;
      }
      activeElement.scrollIntoView({
        block: "nearest",
        inline: "nearest"
      });
    }
  }, [activeState]);
  return {
    multiselect: Boolean(multiselect),
    openState,
    setOpen,
    openKey,
    activeState,
    setActive,
    selectedState,
    setSelectedState,
    select,
    clear,
    focusVisibleState,
    setFocusVisibleState,
    focusedState,
    setFocusedState,
    setListRef,
    listRef,
    options: optionsRef.current.map((option) => option.element),
    register,
    getOptionAtIndex,
    getIndexOfOption,
    getOptionsMatching,
    getOptionFromSearch,
    getOptionAfter,
    getOptionBefore,
    getOptionPageAbove,
    getOptionPageBelow,
    getFirstOption,
    getLastOption,
    valueToString,
    disabled
  };
}

exports.defaultValueToString = defaultValueToString;
exports.useListControl = useListControl;
