// @salt-ds/core/dist-es/form-field/FormFieldLabel.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import '../form-field-context/FormFieldContext.js';
import { useFormFieldProps } from '../form-field-context/useFormFieldProps.js';
import '../text/Code.js';
import '../text/Display.js';
import '../text/Headings.js';
import { Label } from '../text/Label.js';
import '../text/Text.js';
import '../text/TextAction.js';
import '../text/TextNotation.js';
import { capitalize } from '../utils/capitalize.js';
import 'react';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import css_248z from './FormFieldLabel.css.js';

const withBaseName = makePrefixer("saltFormFieldLabel");
const FormFieldLabel = ({
  className,
  children,
  intent = "label",
  onClick,
  ...restProps
}) => {
  const { a11yProps, disabled, necessity, formFieldRef } = useFormFieldProps();
  const targetWindow = useWindow();
  useComponentCssInjection({
    testId: "salt-form-field-label",
    css: css_248z,
    window: targetWindow
  });
  const id = a11yProps == null ? void 0 : a11yProps["aria-labelledby"];
  const handleClick = (event) => {
    if (formFieldRef == null ? void 0 : formFieldRef.current) {
      const element = formFieldRef.current.querySelector(
        `[aria-labelledby*="${id}"]`
      );
      element == null ? void 0 : element.focus();
      element == null ? void 0 : element.click();
    }
    onClick == null ? void 0 : onClick(event);
  };
  const necessityLabel = necessity ? necessity === "asterisk" ? " *" : ` (${capitalize(necessity)})` : void 0;
  return /* @__PURE__ */ jsxs(
    Label,
    {
      className: clsx(withBaseName(), withBaseName(intent), className),
      id,
      disabled,
      variant: "secondary",
      onClick: handleClick,
      ...restProps,
      children: [
        children,
        necessityLabel && /* @__PURE__ */ jsx("span", { className: withBaseName("necessityLabel"), children: necessityLabel })
      ]
    }
  );
};

export { FormFieldLabel };
