// @salt-ds/core/dist-es/number-input/NumberInput.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { useClassNameInjection, useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef, useRef, useState, useEffect } from 'react';
import { Button } from '../button/Button.js';
import '../form-field-context/FormFieldContext.js';
import { useFormFieldProps } from '../form-field-context/useFormFieldProps.js';
import { useIcon } from '../semantic-icon-provider/SemanticIconProvider.js';
import { StatusAdornment } from '../status-adornment/StatusAdornment.js';
import { capitalize } from '../utils/capitalize.js';
import { makePrefixer } from '../utils/makePrefixer.js';
import { useControlled } from '../utils/useControlled.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import { useForkRef } from '../utils/useForkRef.js';
import { useId } from '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { useLongPressPointerAction } from './internal/useLongPressPointerAction.js';
import css_248z from './NumberInput.css.js';

const withBaseName = makePrefixer("saltNumberInput");
const isOutOfRange = (value, min, max) => {
  if (typeof value === "string" && !value.length) {
    return true;
  }
  const floatValue = typeof value === "string" ? Number.parseFloat(value) : value;
  return Number.isNaN(floatValue) || floatValue > max || floatValue < min;
};
function getNumberPrecision(num) {
  const numStr = String(num);
  if (numStr.includes("e") || numStr.includes("E")) {
    const [base, exponent] = numStr.split(/[eE]/);
    const decimalPart = base.split(".")[1] || "";
    const precision = decimalPart.length - Number.parseInt(exponent, 10);
    return Math.max(0, precision);
  }
  if (numStr.includes(".")) {
    return numStr.split(".")[1].length;
  }
  return 0;
}
const defaultPattern = (inputValue) => /^[+-]?(\d+(\.\d*)?|\.\d*)?$/.test(inputValue);
const defaultDecrement = (value, step, stepMultiplier, parse) => {
  const parsedValue = parse(value) ?? 0;
  const decrementStep = step * stepMultiplier;
  return String(parsedValue - decrementStep);
};
const defaultIncrement = (value, step, stepMultiplier, parse) => {
  const parsedValue = parse(value) ?? 0;
  const incrementStep = step * stepMultiplier;
  return String(parsedValue + incrementStep);
};
const defaultFormat = (value, decimalScale) => {
  const sanitized = value.trim();
  if (!sanitized.length) {
    return "";
  }
  const floatValue = Number.parseFloat(sanitized);
  const updatedValue = Number.isNaN(floatValue) ? sanitized : floatValue.toFixed(decimalScale);
  return String(updatedValue);
};
const defaultParse = (value, decimalScale) => {
  const sanitizedValue = value.trim();
  if (!sanitizedValue.length) {
    return null;
  }
  if (sanitizedValue === "." || sanitizedValue === "+" || sanitizedValue === "-") {
    return 0;
  }
  const floatString = Number.parseFloat(value).toFixed(decimalScale);
  return Number.parseFloat(floatString);
};
const NumberInput = forwardRef(
  function NumberInput2(props, ref) {
    const { className, props: finalProps } = useClassNameInjection(
      "saltNumberInput",
      props
    );
    const {
      "aria-valuetext": ariaValueTextProp,
      bordered,
      clamp,
      step = 1,
      stepMultiplier = 2,
      value: valueProp,
      defaultValue,
      decimalScale = Math.max(
        getNumberPrecision(valueProp ?? defaultValue ?? 0),
        getNumberPrecision(step)
      ),
      disabled,
      emptyReadOnlyMarker = "\u2014",
      endAdornment,
      format = (v) => defaultFormat(v, decimalScale),
      hideButtons,
      id,
      pattern = defaultPattern,
      inputProps = {},
      inputRef: inputRefProp,
      max = Number.MAX_SAFE_INTEGER,
      min = Number.MIN_SAFE_INTEGER,
      onBlur,
      onChange,
      onMouseUp,
      onNumberChange,
      parse = (v) => defaultParse(v, decimalScale),
      placeholder,
      readOnly,
      startAdornment,
      decrement = (v, s, m) => defaultDecrement(v, s, m, parse),
      increment = (v, s, m) => defaultIncrement(v, s, m, parse),
      textAlign = "left",
      validationStatus: validationStatusProp,
      variant = "primary",
      ...restProps
    } = finalProps;
    const targetWindow = useWindow();
    useComponentCssInjection({
      testId: "salt-number-input",
      css: css_248z,
      window: targetWindow
    });
    const {
      a11yProps: {
        "aria-describedby": formFieldDescribedBy,
        "aria-labelledby": formFieldLabelledBy
      } = {},
      disabled: formFieldDisabled,
      readOnly: formFieldReadOnly,
      necessity: formFieldRequired,
      validationStatus: formFieldValidationStatus
    } = useFormFieldProps();
    const isDisabled = disabled || formFieldDisabled;
    const isReadOnly = readOnly || formFieldReadOnly;
    const validationStatus = formFieldValidationStatus ?? validationStatusProp;
    const validationStatusId = useId(id);
    const inputRef = useRef(null);
    const handleInputRef = useForkRef(inputRefProp, inputRef);
    const {
      "aria-describedby": inputDescribedBy,
      "aria-labelledby": inputLabelledBy,
      className: inputClassName,
      onBlur: inputOnBlur,
      onFocus: inputOnFocus,
      required: inputRequired,
      onKeyDown: inputOnKeyDown,
      ...restInputProps
    } = inputProps;
    const isRequired = formFieldRequired ? ["required", "asterisk"].includes(formFieldRequired) : inputRequired;
    const [isFocused, setIsFocused] = useState(false);
    const [isEditing, setIsEditing] = useState(false);
    const { DecreaseIcon, IncreaseIcon } = useIcon();
    const [value, setValue] = useControlled({
      controlled: valueProp !== void 0 ? String(valueProp) : void 0,
      default: String(defaultValue ?? ""),
      name: "NumberInput",
      state: "value"
    });
    const lastCommitValue = useRef(value);
    const commit = (event, newNumber, newInputValue) => {
      let safeNumber = newNumber;
      if (safeNumber !== null && !Number.isNaN(safeNumber)) {
        safeNumber = Math.max(
          Number.MIN_SAFE_INTEGER,
          Math.min(Number.MAX_SAFE_INTEGER, safeNumber)
        );
        if (clamp) {
          safeNumber = Math.max(min, Math.min(max, safeNumber));
        }
      } else {
        safeNumber = null;
      }
      const commitValue = safeNumber !== null ? format(String(safeNumber)) : newInputValue;
      if (commitValue !== value) {
        setValue(commitValue);
        onChange == null ? void 0 : onChange(event, commitValue);
      }
      if (lastCommitValue.current !== commitValue) {
        onNumberChange == null ? void 0 : onNumberChange(event, safeNumber);
        lastCommitValue.current = commitValue;
      }
    };
    const handleBlur = (event) => {
      setIsFocused(false);
      onBlur == null ? void 0 : onBlur(event);
    };
    const handleInputBlur = (event) => {
      setIsEditing(false);
      inputOnBlur == null ? void 0 : inputOnBlur(event);
      const parsedValue = parse(value);
      commit(event, parsedValue, value);
    };
    const handleInputFocus = (event) => {
      setIsEditing(false);
      inputOnFocus == null ? void 0 : inputOnFocus(event);
    };
    const handleInputChange = (event) => {
      const inputValue = event.currentTarget.value;
      if (!inputValue.length) {
        setValue("");
        onChange == null ? void 0 : onChange(event, "");
        return;
      }
      const validValue = pattern ? pattern(inputValue) : true;
      if (validValue) {
        setIsEditing(true);
        onChange == null ? void 0 : onChange(event, event.target.value);
        setValue(inputValue);
      } else {
        event.preventDefault();
      }
    };
    let floatValue = parse(value) ?? 0;
    floatValue = Math.max(
      Number.MIN_SAFE_INTEGER,
      Math.min(Number.MAX_SAFE_INTEGER, floatValue)
    );
    if (clamp) {
      floatValue = Math.max(min, Math.min(max, floatValue));
    }
    const decrementValue = (event, block) => {
      const validValue = parse(value) ?? 0;
      if (Number.isNaN(validValue)) {
        return;
      }
      const newValue = decrement(
        String(validValue),
        step,
        block ? stepMultiplier : 1
      );
      const parsedValue = parse(newValue);
      commit(event ?? null, parsedValue, newValue);
    };
    const activateDecrement = useLongPressPointerAction(
      decrementValue,
      floatValue <= min
    );
    const incrementValue = (event, block) => {
      const validValue = parse(value) ?? 0;
      if (Number.isNaN(validValue)) {
        return;
      }
      const newValue = increment(
        String(validValue),
        step,
        block ? stepMultiplier : 1
      );
      const parsedValue = parse(newValue);
      commit(event ?? null, parsedValue, newValue);
    };
    const activateIncrement = useLongPressPointerAction(
      incrementValue,
      floatValue >= max
    );
    useEffect(() => {
      var _a;
      if (isFocused) {
        (_a = inputRef.current) == null ? void 0 : _a.focus();
      }
    }, [isFocused]);
    const handleInputKeyDown = (event) => {
      switch (event.key) {
        case "ArrowUp": {
          event.preventDefault();
          const block = event.shiftKey;
          incrementValue(event, block);
          break;
        }
        case "ArrowDown": {
          event.preventDefault();
          const block = event.shiftKey;
          decrementValue(event, block);
          break;
        }
        case "Home": {
          event.preventDefault();
          const newValue = String(min);
          commit(event, min, newValue);
          break;
        }
        case "End": {
          event.preventDefault();
          const newValue = String(max);
          commit(event, max, newValue);
          break;
        }
        case "PageUp": {
          event.preventDefault();
          incrementValue(event, true);
          break;
        }
        case "PageDown": {
          event.preventDefault();
          decrementValue(event, true);
          break;
        }
      }
      inputOnKeyDown == null ? void 0 : inputOnKeyDown(event);
    };
    const handleIncrementPointerDown = (event, disableIncrement2) => {
      event.preventDefault();
      if (!disableIncrement2) {
        setIsEditing(false);
        activateIncrement(event);
      } else if (inputRef.current) {
        inputRef.current.select();
      }
    };
    const handleDecrementPointerDown = (event, disableDecrement2) => {
      event.preventDefault();
      if (!disableDecrement2) {
        setIsEditing(false);
        activateDecrement(event);
      } else if (inputRef.current) {
        inputRef.current.select();
      }
    };
    const handleContainerMouseUp = (event) => {
      setIsFocused(true);
      onMouseUp == null ? void 0 : onMouseUp(event);
    };
    let renderedValue;
    if (isEditing) {
      renderedValue = value;
    } else if (!(value == null ? void 0 : value.length)) {
      renderedValue = "";
    } else {
      renderedValue = format(
        Number.isNaN(floatValue) ? value : String(floatValue)
      );
    }
    const disableDecrement = disabled || floatValue - step < min;
    const disableIncrement = disabled || floatValue + step > max;
    return /* @__PURE__ */ jsxs(
      "div",
      {
        className: clsx(
          withBaseName(),
          withBaseName(variant),
          {
            [withBaseName("focused")]: isFocused,
            [withBaseName("disabled")]: isDisabled,
            [withBaseName("readOnly")]: isReadOnly,
            [withBaseName("hiddenButtons")]: hideButtons,
            [withBaseName(validationStatus || "")]: validationStatus,
            [withBaseName("bordered")]: bordered
          },
          className
        ),
        onBlur: handleBlur,
        onMouseUp: handleContainerMouseUp,
        ...restProps,
        ref,
        children: [
          startAdornment && /* @__PURE__ */ jsx("div", { className: withBaseName("startAdornmentContainer"), children: startAdornment }),
          /* @__PURE__ */ jsx(
            "input",
            {
              "aria-describedby": clsx(formFieldDescribedBy, inputDescribedBy) || void 0,
              "aria-labelledby": clsx(formFieldLabelledBy, inputLabelledBy) || void 0,
              "aria-invalid": !isReadOnly && renderedValue.length ? isOutOfRange(floatValue, min, max) || validationStatus === "error" : void 0,
              className: clsx(
                withBaseName("input"),
                withBaseName(`inputTextAlign${capitalize(textAlign)}`),
                inputClassName
              ),
              disabled: isDisabled,
              onBlur: handleInputBlur,
              onChange: handleInputChange,
              onFocus: handleInputFocus,
              onKeyDown: isReadOnly ? void 0 : handleInputKeyDown,
              placeholder,
              readOnly: isReadOnly,
              "aria-readonly": isReadOnly ? "true" : void 0,
              ref: handleInputRef,
              required: isRequired,
              "aria-valuemax": !isReadOnly && renderedValue.length ? max : void 0,
              "aria-valuemin": !isReadOnly && renderedValue.length ? min : void 0,
              "aria-valuenow": !isReadOnly ? floatValue : void 0,
              "aria-valuetext": !isReadOnly ? renderedValue.length ? ariaValueTextProp ?? renderedValue : "Empty" : void 0,
              role: isReadOnly ? "textbox" : "spinbutton",
              tabIndex: isDisabled ? -1 : 0,
              value: isReadOnly && renderedValue.length === 0 ? emptyReadOnlyMarker : renderedValue,
              ...restInputProps
            }
          ),
          /* @__PURE__ */ jsx("div", { className: withBaseName("activationIndicator") }),
          !isDisabled && validationStatus && /* @__PURE__ */ jsx(StatusAdornment, { status: validationStatus, id: validationStatusId }),
          endAdornment && /* @__PURE__ */ jsx("div", { className: withBaseName("endAdornmentContainer"), children: endAdornment }),
          !isReadOnly && /* @__PURE__ */ jsxs("div", { className: clsx(withBaseName("buttonContainer")), children: [
            /* @__PURE__ */ jsx(
              Button,
              {
                "aria-hidden": true,
                appearance: "transparent",
                tabIndex: -1,
                disabled: disableIncrement,
                className: withBaseName("increment"),
                onPointerDown: (event) => handleIncrementPointerDown(event, disableIncrement),
                children: /* @__PURE__ */ jsx(IncreaseIcon, { "aria-hidden": true })
              }
            ),
            /* @__PURE__ */ jsx(
              Button,
              {
                "aria-hidden": true,
                appearance: "transparent",
                tabIndex: -1,
                disabled: disableDecrement,
                className: withBaseName("decrement"),
                onPointerDown: (event) => handleDecrementPointerDown(event, disableDecrement),
                children: /* @__PURE__ */ jsx(DecreaseIcon, { "aria-hidden": true })
              }
            )
          ] })
        ]
      }
    );
  }
);

export { NumberInput, isOutOfRange };
