// @salt-ds/core/dist-cjs/table/Table.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
var makePrefixer = require('../utils/makePrefixer.js');
var useIsomorphicLayoutEffect = require('../utils/useIsomorphicLayoutEffect.js');
require('../utils/useFloatingUI/useFloatingUI.js');
var useId = require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var Table$1 = require('./Table.css.js');
var TableContext = require('./TableContext.js');

const withTableBaseName = makePrefixer.makePrefixer("saltTable");
const Table = React.forwardRef(
  function Table2(props, ref) {
    const targetWindow = window.useWindow();
    styles.useComponentCssInjection({
      testId: "salt-table",
      css: Table$1,
      window: targetWindow
    });
    const {
      children,
      className,
      variant = "primary",
      divider = "tertiary",
      zebra = false,
      "aria-labelledby": ariaLabelledBy,
      id,
      ...rest
    } = props;
    const tableId = useId.useId(id);
    const { setId, setLabelledBy } = TableContext.useTable();
    const labelledBy = ariaLabelledBy ?? tableId;
    useIsomorphicLayoutEffect.useIsomorphicLayoutEffect(() => {
      if (tableId) {
        setId(tableId);
      }
      if (labelledBy) {
        setLabelledBy(labelledBy);
      }
    }, [tableId, labelledBy, setId, setLabelledBy]);
    return /* @__PURE__ */ jsxRuntime.jsx(
      "table",
      {
        id: tableId,
        className: clsx.clsx(
          withTableBaseName(),
          {
            [withTableBaseName(variant)]: variant,
            [withTableBaseName("zebra")]: zebra,
            [withTableBaseName(`divider-${divider}`)]: divider
          },
          className
        ),
        "aria-labelledby": labelledBy,
        ref,
        ...rest,
        children
      }
    );
  }
);

exports.Table = Table;
exports.withTableBaseName = withTableBaseName;
