// @salt-ds/core/dist-cjs/table/TableContainer.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
var useIsomorphicLayoutEffect = require('../utils/useIsomorphicLayoutEffect.js');
require('../utils/useFloatingUI/useFloatingUI.js');
var useForkRef = require('../utils/useForkRef.js');
require('../utils/useId.js');
var useResizeObserver = require('../utils/useResizeObserver.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var Table$1 = require('./Table.js');
var Table = require('./Table.css.js');
var TableContext = require('./TableContext.js');

const TableContainer = React.forwardRef(
  function TableContainer2(props, ref) {
    const [isOverflowing, setIsOverflowing] = React.useState(false);
    const [tableId, setTableId] = React.useState();
    const [labelledBy, setLabelledBy] = React.useState();
    const targetWindow = window.useWindow();
    styles.useComponentCssInjection({
      testId: "salt-table-container",
      css: Table,
      window: targetWindow
    });
    const {
      children,
      className,
      // @ts-expect-error: "aria-labelledby" is omitted to prevent accidental misuse,
      // but we still want to forward it for advanced accessible labeling scenarios.
      "aria-labelledby": ariaLabelledBy,
      // @ts-expect-error: Allow passing aria-label even though it's omitted from HTMLAttributes
      // Same reasoning as above: we forward aria-label for accessibility purposes.
      "aria-label": ariaLabel,
      // @ts-expect-error: Allow passing role even though it's omitted from HTMLAttributes
      // Same reasoning as above: we forward role for accessibility purposes.
      role,
      tabIndex,
      ...rest
    } = props;
    const scrollRef = React.useRef(null);
    const handleRef = useForkRef.useForkRef(ref, scrollRef);
    const checkOverflow = React.useCallback(() => {
      const element = scrollRef.current;
      if (!element) return;
      const overflowing = element.scrollHeight > element.clientHeight || element.scrollWidth > element.clientWidth;
      setIsOverflowing(overflowing);
    }, []);
    useResizeObserver.useResizeObserver({ ref: scrollRef, onResize: checkOverflow });
    useIsomorphicLayoutEffect.useIsomorphicLayoutEffect(() => {
      checkOverflow();
    }, [checkOverflow]);
    const overflowProps = isOverflowing ? {
      role: role ?? "region",
      tabIndex: tabIndex ?? 0,
      "aria-label": ariaLabel,
      ...ariaLabelledBy === void 0 && ariaLabel === void 0 && {
        "aria-labelledby": labelledBy ?? tableId
      },
      ...ariaLabelledBy !== void 0 && {
        "aria-labelledby": ariaLabelledBy
      }
    } : {
      role,
      tabIndex,
      "aria-labelledby": ariaLabelledBy,
      "aria-label": ariaLabel
    };
    const contextValue = React.useMemo(
      () => ({ id: tableId, setId: setTableId, labelledBy, setLabelledBy }),
      [tableId, labelledBy]
    );
    return /* @__PURE__ */ jsxRuntime.jsx(TableContext.TableContext.Provider, { value: contextValue, children: /* @__PURE__ */ jsxRuntime.jsx(
      "div",
      {
        ref: handleRef,
        className: clsx.clsx(Table$1.withTableBaseName("container"), className),
        ...overflowProps,
        ...rest,
        children
      }
    ) });
  }
);

exports.TableContainer = TableContainer;
