// @salt-ds/core/dist-cjs/table/THead.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');
var clsx = require('clsx');
var React = require('react');
var Table$1 = require('./Table.js');
var Table = require('./Table.css.js');

const THead = React.forwardRef(
  function THead2({
    children,
    className,
    sticky = false,
    variant,
    divider = "primary",
    ...rest
  }, ref) {
    const targetWindow = window.useWindow();
    styles.useComponentCssInjection({
      testId: "salt-table-thead",
      css: Table,
      window: targetWindow
    });
    return /* @__PURE__ */ jsxRuntime.jsx(
      "thead",
      {
        ref,
        className: clsx.clsx(
          Table$1.withTableBaseName("thead"),
          { [Table$1.withTableBaseName(`thead-${variant}`)]: variant },
          { [Table$1.withTableBaseName("thead-sticky")]: sticky },
          { [Table$1.withTableBaseName(`thead-divider-${divider}`)]: divider },
          className
        ),
        ...rest,
        children
      }
    );
  }
);

exports.THead = THead;
