// @salt-ds/core/dist-es/collapsible/Collapsible.js
import { jsx } from 'react/jsx-runtime';
import { clsx } from 'clsx';
import { forwardRef, useState, useCallback, useMemo } from 'react';
import { makePrefixer } from '../utils/makePrefixer.js';
import { useControlled } from '../utils/useControlled.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { CollapsibleContext } from './CollapsibleContext.js';

const withBaseName = makePrefixer("saltCollapsible");
const Collapsible = forwardRef(
  function Collapsible2(props, ref) {
    const {
      className,
      open: openProp,
      defaultOpen,
      onOpenChange,
      ...rest
    } = props;
    const [open, setOpenState] = useControlled({
      default: Boolean(defaultOpen),
      controlled: openProp,
      name: "Collapsible",
      state: "open"
    });
    const [panelId, setPanelId] = useState(void 0);
    const setOpen = useCallback(
      (event, newOpen) => {
        setOpenState(newOpen);
        onOpenChange == null ? void 0 : onOpenChange(event, newOpen);
      },
      [onOpenChange]
    );
    const contextValue = useMemo(
      () => ({
        open,
        setOpen,
        panelId,
        setPanelId
      }),
      [open, setOpen, panelId]
    );
    return /* @__PURE__ */ jsx(CollapsibleContext.Provider, { value: contextValue, children: /* @__PURE__ */ jsx("div", { className: clsx(withBaseName(), className), ref, ...rest }) });
  }
);

export { Collapsible };
