// @salt-ds/core/dist-es/collapsible/CollapsiblePanel.js
import { jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef, useEffect } from 'react';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import { useId } from '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { useCollapsibleContext } from './CollapsibleContext.js';
import css_248z from './CollapsiblePanel.css.js';

const withBaseName = makePrefixer("saltCollapsiblePanel");
const CollapsiblePanel = forwardRef((props, ref) => {
  const { children, className, id: idProp, ...rest } = props;
  const targetWindow = useWindow();
  useComponentCssInjection({
    testId: "salt-collapsible-panel",
    css: css_248z,
    window: targetWindow
  });
  const id = useId(idProp);
  const { open, setPanelId } = useCollapsibleContext();
  useEffect(() => {
    if (id) {
      setPanelId == null ? void 0 : setPanelId(id);
    }
  }, [id, setPanelId]);
  return /* @__PURE__ */ jsx(
    "div",
    {
      className: clsx(withBaseName(), className),
      id,
      "aria-hidden": !open ? "true" : void 0,
      hidden: !open,
      ref,
      ...rest,
      children: /* @__PURE__ */ jsx("div", { className: withBaseName("inner"), children })
    }
  );
});

export { CollapsiblePanel };
