// @salt-ds/core/dist-cjs/tooltip/Tooltip.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var styles = require('@salt-ds/styles');
var clsx = require('clsx');
var React = require('react');
require('../form-field-context/FormFieldContext.js');
var useFormFieldProps = require('../form-field-context/useFormFieldProps.js');
var getRefFromChildren = require('../utils/getRefFromChildren.js');
var makePrefixer = require('../utils/makePrefixer.js');
var mergeProps = require('../utils/mergeProps.js');
var useFloatingUI = require('../utils/useFloatingUI/useFloatingUI.js');
var useForkRef = require('../utils/useForkRef.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var TooltipBase = require('./TooltipBase.js');
var useTooltip = require('./useTooltip.js');

const withBaseName = makePrefixer.makePrefixer("saltTooltip");
const Tooltip = React.forwardRef(
  function Tooltip2(props, ref) {
    const { className, props: finalProps } = styles.useClassNameInjection(
      "saltTooltip",
      props
    );
    const {
      children,
      disabled: disabledProp = false,
      hideArrow = false,
      hideIcon = false,
      open: openProp,
      content,
      status: statusProp,
      placement = "right",
      enterDelay = 300,
      leaveDelay = 0,
      ...rest
    } = finalProps;
    const {
      disabled: formFieldDisabled,
      validationStatus: formFieldValidationStatus
    } = useFormFieldProps.useFormFieldProps();
    const disabled = disabledProp || formFieldDisabled;
    const status = statusProp ?? (formFieldValidationStatus || void 0);
    const { Component: FloatingComponent } = useFloatingUI.useFloatingComponent();
    const hookProps = {
      open: openProp,
      disabled,
      placement,
      enterDelay,
      leaveDelay,
      ...rest
    };
    const {
      arrowProps,
      open,
      floating,
      reference,
      getTriggerProps,
      getTooltipProps,
      getTooltipPosition
    } = useTooltip.useTooltip(hookProps);
    const triggerRef = useForkRef.useForkRef(getRefFromChildren.getRefFromChildren(children), reference);
    const floatingRef = useForkRef.useForkRef(floating, ref);
    const hasContent = content != null && content !== "";
    return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
      React.isValidElement(children) && React.cloneElement(children, {
        ...mergeProps.mergeProps(getTriggerProps(), children.props),
        ref: triggerRef
      }),
      /* @__PURE__ */ jsxRuntime.jsx(
        FloatingComponent,
        {
          className: clsx.clsx(
            withBaseName(),
            { [withBaseName(status ?? "")]: status },
            className
          ),
          open: open && !disabled && hasContent,
          ...getTooltipProps(),
          ref: floatingRef,
          ...getTooltipPosition(),
          children: /* @__PURE__ */ jsxRuntime.jsx(
            TooltipBase.TooltipBase,
            {
              hideIcon,
              status,
              content,
              hideArrow,
              arrowProps
            }
          )
        }
      )
    ] });
  }
);

exports.Tooltip = Tooltip;
