// @salt-ds/core/dist-es/kbd/Kbd.js
import { jsx } from 'react/jsx-runtime';
import { useComponentCssInjection } from '@salt-ds/styles';
import { useWindow } from '@salt-ds/window';
import { clsx } from 'clsx';
import { forwardRef } from 'react';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import css_248z from './Kbd.css.js';

const withBaseName = makePrefixer("saltKbd");
const Kbd = forwardRef(
  function Kbd2(props, ref) {
    const { children, className, variant = "primary", ...rest } = props;
    const targetWindow = useWindow();
    useComponentCssInjection({
      testId: "salt-kbd",
      css: css_248z,
      window: targetWindow
    });
    return /* @__PURE__ */ jsx(
      "kbd",
      {
        ref,
        className: clsx(withBaseName(), withBaseName(variant), className),
        ...rest,
        children
      }
    );
  }
);

export { Kbd };
